---
title: "Untitled"
author: "Nicholas Dove"
date: "4/7/2021"
output: html_document
---
# Set libraries
```{r, warning=F, message=F, cache=F, results= F}
library(tidyverse)
library(phyloseq)
library(readxl)
library(ggplot2)
library(cowplot)
library(biomformat)
library(yaml)
library(parallel)
library(doParallel)
library(microbiome)
library(hillR)
library(car)

#library(metagMisc)
#library(mixOmics)

source("~/Downloads/qza_to_phyloseq.R")

#Set themes -------------------------------------
theme_set(theme_cowplot())

theme_update(axis.title = element_text(size = 12), axis.text = element_text(size = 10), strip.text.x = element_text(size = 10),
             strip.text.y = element_text(size = 10),
             legend.position = "none", panel.grid.major = element_blank(), 
             panel.grid.minor = element_blank())
```

# Pathogen abundnace across genotypes and infections (Figure 1)
```{r, warning=F, message=F, cache=F, results= F}
#Download data
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 

# merge data
data <- merge(sep, metadata, by = "SampleID")

#Remove samples
data <- data %>% filter(SampleID != "S71" & SampleID != "S60" & SampleID != "S62")

# convert data to copy per mg
data_LE <- data %>%
  filter(sample_type == "LE") %>%
  mutate(sep = Copy_number_per_ul*30/50)
data_LW <- data %>%
  filter(sample_type == "LW") %>%
  mutate(sep = Copy_number_per_ul*50/1000)
data_RE <- data %>%
  filter(sample_type == "RE") %>%
  mutate(sep = Copy_number_per_ul*30/50)
data_RH <- data %>%
  filter(sample_type == "RH") %>%
  mutate(sep = Copy_number_per_ul*50/250)
data_new <- rbind(data_LE, data_LW, data_RE, data_RH)

points <- data_new %>%
  dplyr::mutate(Genotype = ordered(Genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148")))

data_new %>%
  dplyr::mutate(Genotype = ordered(Genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(Genotype, sample_type, resistance) %>%
  dplyr::summarise(mean = mean(sep), se = sd(sep)/sqrt(n())) %>%
  ggplot(aes(x = Genotype, y = mean, fill = resistance)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  #geom_jitter(data = points, aes(x = Genotype, y = sep, fill = resistance), pch = 21, color = "black", alpha = 0.5, inherit.aes = F) +
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote(atop(italic("S. musiva")~"abundance", "(copy"~mg^-1*")"))) +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), labels = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  facet_grid(~sample_type, scales = "free", 
             labeller = labeller(sample_type = c("LE" = "Leaf Endosphere", "LW" = "Leaf Surface", "RE" = "Root Endosphere", "RH" = "Rhizosphere"))) +
  panel_border() +
  theme(axis.text.x = element_text(angle = 45, size = 9, hjust = 1), legend.title = element_blank(), legend.background = element_rect(fill = "white"),
        legend.position = c(0.02, 0.85), legend.text = element_text(size = 12)) -> fig.1a

#data_new$pointx <- as.numeric(as.factor(data_new$resistance)) - 0.25 + (data_new$Infection_2020)/2

data_new %>% 
  group_by(sample_type, resistance, Infection_2020) %>%
  dplyr::summarise(mean = mean(sep), se = sd(sep)/sqrt(n())) %>%
  rbind(data.frame(mean = rep(0, 4), se = rep(0, 4), resistance = rep("deltoides", 4), 
                   Infection_2020 = rep(1, 4), sample_type = c("LE", "LW", "RE", "RH"))) %>%
  ggplot(aes(x = resistance, y = mean, fill = as.character(Infection_2020))) +
  geom_bar(stat = "identity", position = position_dodge(0.9), color = "black") +
  #geom_jitter(data = data_new, aes(x = pointx, y = sep, fill = as.factor(as.character(Infection_2020))), 
  #            pch = 21, color = "black", alpha = 0.5, inherit.aes = F) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2, position = position_dodge(0.9)) +
  scale_y_continuous(name = bquote(atop(italic("S. musiva")~"abundance", "(copy"~mg^-1*")"))) +
  scale_x_discrete(name = NULL, labels = c("deltoides", "trichocarpa\n(tolerant)", "trichocarpa\n(susceptible)")) +
  scale_fill_manual(values = c("grey", "#8B0000"), labels = c("noninfected", "infected"))  +
  facet_grid(~sample_type, scales = "free", labeller = labeller(sample_type = c("LE" = "Leaf Endosphere", "LW" = "Leaf Surface", "RE" = "Root Endosphere", "RH" = "Rhizosphere"))) +
  panel_border() +
  theme(axis.text.x = element_text(angle = 45, size = 9, hjust = 1), legend.title = element_blank(), 
        legend.position = c(0.02, 0.85), legend.text = element_text(size = 12)) -> fig.1b

fig1 <- plot_grid(fig.1a, fig.1b, align = "vh", labels = "AUTO", ncol = 1)

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure1.pdf", width = 7.5, height = 8, units = "in", dpi = 600)
fig1
dev.off()

fig.1b +
  theme(axis.text = element_text(size = 24), axis.title = element_text(size = 28), strip.text.x = element_text(size = 24),
        axis.text.x = element_text(size = 24, angle = 45, hjust = 1),
        strip.background = element_rect(color = "black"), legend.text = element_text(size = 24)) +
  panel_border(color = "black") -> p

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure1b_poster.pdf", width = 13, height = 7, units = "in", dpi = 600)
p
dev.off()

# Stats ----------------------------------------------
## Septoria by Genotype/resistance -------
a <- aov(log(sep) ~ sample_type, data_new)
summary(a) # Only sample type sig.

a <- kruskal_test(sep ~ sample_type, data = data_new)

a <- aov(log(sep) ~ sample_type*Genotype, data_new)
summary(a) # Only sample type sig.
data_new$species <- "trich"
data_new$species[data_new$resistance == "deltoides"] <- "delt"
a <- aov(log(sep) ~ sample_type*species, data_new)
summary(a) # Only sample type sig.
a <- aov(log(sep) ~ sample_type*resistance, data_new %>% filter(species == "trich"))
summary(a) # Only sample type sig.

for(i in levels(as.factor(data_new$sample_type))){
  sub <- subset(data_new, sample_type %in% i)
  print(i)
  a <- aov(log(sep) ~ resistance, sub %>% filter(species == "trich"))
  print(p.adjust(summary(a)[[1]][[1,"Pr(>F)"]], method = "fdr", n = 4))
} # none sig.



for(i in levels(as.factor(data_new$sample_type))){
  sub <- subset(data_new, sample_type %in% i)
  print(i)
  a <- aov(log(sep) ~ Genotype, sub)
  print(p.adjust(summary(a)[[1]][[1,"Pr(>F)"]], method = "fdr", n = 4))
} # none sig.

for(i in levels(as.factor(data_new$sample_type))){
  sub <- subset(data_new, sample_type %in% i)
  print(i)
  a <- aov(log(sep) ~ resistance, sub)
  print(p.adjust(summary(a)[[1]][[1,"Pr(>F)"]], method = "fdr", n = 4))
} # none sig.

##Septoria by host species
data_new$species <- "trich"
data_new$species[data_new$resistance == "deltoides"] <- "delt"
a <- aov(log(sep) ~ species*sample_type, data_new)
summary(a) # Only sample type sig.
for(i in levels(as.factor(data_new$sample_type))){
  sub <- subset(data_new, sample_type %in% i)
  print(i)
  a <- aov(log(sep) ~ species, sub)
  print(p.adjust(summary(a)[[1]][[1,"Pr(>F)"]], method = "fdr", n = 4))
} # none sig.

## Septoria by infection/resistance ------------

a <- glm(as.factor(as.character(Infection_2020)) ~ log(sep)*resistance*sample_type, data = data_new %>% filter(resistance != "deltoides"), family = "binomial")

for(i in levels(as.factor(data_new$sample_type))){
  sub <- subset(data_new, sample_type %in% i)
  print(i)
  a <- t.test(sep ~as.factor(as.character(Infection_2020)), data =sub %>% filter(resistance != "deltoides"))
  print(a)
} # Infection never sig.

a <- aov(log(sep) ~ as.factor(as.character(Infection_2020))*resistance*sample_type, data_new %>% filter(resistance != "deltoides"))
summary(a) # resistance*sample type interaction
for(i in levels(as.factor(data_new$sample_type))){
  sub <- subset(data_new, sample_type %in% i)
  print(i)
  a <- aov(log(sep) ~ resistance*as.factor(as.character(Infection_2020)), sub %>% filter(resistance != "deltoides"))
  print(summary(a))
} # Infection never sig.


# Correlations among tissues
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 

# merge data
data <- merge(sep, metadata, by = "SampleID")

# convert data to copy per mg
data_LE <- data %>%
  filter(sample_type == "LE") %>%
  mutate(sep_LE = Copy_number_per_ul*30/50)
data_LW <- data %>%
  filter(sample_type == "LW") %>%
  mutate(sep_LW = Copy_number_per_ul*50/1000)
data_RE <- data %>%
  filter(sample_type == "RE") %>%
  mutate(sep_RE = Copy_number_per_ul*30/50)
data_RH <- data %>%
  filter(sample_type == "RH") %>%
  mutate(sep_RH = Copy_number_per_ul*50/250)
data <- merge(merge(merge(data_LE, data_LW, by = "SampleID"), data_RE,by = "SampleID"), data_RH, by = "SampleID")

cor.test(data$sep_LE, data$sep_LW)
cor.test(data$sep_LE, data$sep_LW, method = "spearman")
cor.test(data$sep_LE, data$sep_RE)
cor.test(data$sep_LE, data$sep_RE, method = "spearman")
cor.test(data$sep_LE, data$sep_RH)
cor.test(data$sep_LE, data$sep_RH, method = "spearman")
cor.test(data$sep_LW, data$sep_RE)
cor.test(data$sep_LW, data$sep_RE, method = "spearman")
cor.test(data$sep_LW, data$sep_RH) # only one that is significant
cor.test(data$sep_LW, data$sep_RH, method = "spearman")
cor.test(data$sep_RE, data$sep_RH)
cor.test(data$sep_RE, data$sep_RH, method = "spearman")
```

# Metabolites
```{r, warning=F, message=F, cache=F, results= F}
metabolites_root <- read_excel("~/Downloads/RootMetabolites.xlsx")
metabolites_root <- filter(metabolites_root, SampleID != "S71" & SampleID != "S60" & SampleID != "S62")
metadata <- read_excel("~/Downloads/Scoring.xlsx") 
sep <- read_delim("~/Downloads/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "RE")
metabolites_root <- merge(metabolites_root, 
                     metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T) %>%
  mutate(sample_type = "root")
metabolites_root <- merge(metabolites_root, sep %>% dplyr::select(Copy_number_per_ul, SampleID), by = "SampleID", all.x = T)

metabolites_leaf <- read_excel("~/Downloads/LeafMetabolites.xlsx")
metabolites_leaf <- filter(metabolites_leaf, SampleID != "S62" & SampleID != "S60" & SampleID != "S10")
sep <- read_delim("~/Downloads/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "LE")
metabolites_leaf <- merge(metabolites_leaf, 
                     metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T) %>%
  mutate(sample_type = "leaf")
metabolites_leaf <- merge(metabolites_leaf, sep %>% dplyr::select(Copy_number_per_ul, SampleID), by = "SampleID", all.x = T)

metabolites <- bind_rows(metabolites_root, metabolites_leaf)

metabolites$species <- "trich"
metabolites$species[metabolites$resistance == "deltoides"] <- "delt"

# Species effect
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="leaf") %>%
                      dplyr::select(-Genotype, -resistance, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-species) ~ sub$species, na.rm = T)
ad
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="root") %>%
                      dplyr::select(-Genotype, -resistance, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-species) ~ sub$species, na.rm = T)
ad

#Genotype effect
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="leaf" & species == "trich") %>%
                      dplyr::select(-resistance, -species, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-Genotype) ~ sub$Genotype, na.rm = T)
ad
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="root" & species == "trich") %>%
                      dplyr::select(-resistance, -species, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-Genotype) ~ sub$Genotype, na.rm = T)
ad # sig. but small

#resistance effect
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="leaf" & species == "trich") %>%
                      dplyr::select(-Genotype, -species, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-resistance) ~ sub$resistance, na.rm = T)
ad
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="root" & species == "trich") %>%
                      dplyr::select(-Genotype, -species, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-resistance) ~ sub$resistance, na.rm = T)
ad # sig. but small

#infection effect
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="leaf" & species == "trich") %>%
                      dplyr::select(-Genotype, -species, -SampleID, -resistance, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-Infection_2020) ~ sub$Infection_2020, na.rm = T)
ad # sig. but small
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="leaf" & species == "trich") %>%
                      dplyr::select(-Genotype, -species, -SampleID, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-Infection_2020, -resistance) ~ sub$Infection_2020*sub$resistance, na.rm = T)
ad # sig. but small -- does not relate to resistance
set.seed(01221990)
sub <- metabolites %>% filter(sample_type =="root" & species == "trich") %>%
                      dplyr::select(-Genotype, -species, -SampleID, -resistance, -sample_type, -SampleID, -Copy_number_per_ul)
ad <- vegan::adonis(sub %>% dplyr::select(-Infection_2020) ~ sub$Infection_2020, na.rm = T)
ad # nope!

# Are any metabolites indicative of infection?
# Across habitats
metabolite_long <- metabolites %>%
  gather(key = "metabolite", value = "value", -Genotype, -resistance, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul, -species)

tab <- data.frame(matrix(ncol = 2))
colnames(tab) <- c("metabolite", "p")
metabolite_long_clean <- metabolite_long %>% 
  na.omit() %>% 
  filter(metabolite != "17.16 363 273 257" & metabolite != "14.70 284 269 glucoside" & metabolite != "9.82 269 284" & metabolite != "HCH-deltoidin")
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  a <- t.test(value ~ as.factor(as.character(Infection_2020)), data = metabolite_long %>% filter(metabolite %in% i & resistance != "deltoides"), family = "binomial")
  p <- a$p.value
  tab <- rbind(tab, c(i, round(p, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_t <- tab %>% filter(p.adj < 0.05)
sig_tab_t <- tab %>% filter(p < 0.05)

for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long %>% filter(metabolite %in% i & resistance != "deltoides")
  a <- wilcox.test(x = sub$value, y = sub$Infection_2020)
  p <- a$p.value
  tab <- rbind(tab, c(i, round(p, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_t <- tab %>% filter(p.adj < 0.05)
sig_tab_t <- tab %>% filter(p < 0.05)

# Within only leaves
tab <- data.frame(matrix(ncol = 2))
colnames(tab) <- c("metabolite", "p")
metabolite_long_clean <- metabolite_long %>% 
  filter(sample_type == "leaf") %>% 
  na.omit() %>% 
  filter(metabolite != "17.16 363 273 257" & metabolite != "14.70 284 269 glucoside" & metabolite != "9.82 269 284" & metabolite != "HCH-deltoidin")
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  a <- t.test(value ~ as.factor(as.character(Infection_2020)), data = metabolite_long %>% filter(sample_type == "leaf" & metabolite %in% i & resistance != "deltoides"), family = "binomial")
  p <- a$p.value
  tab <- rbind(tab, c(i, round(p, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(metabolite_long_clean$metabolite))))
sig_tab_leaves_t <- tab %>% filter(p.adj < 0.05)

tab <- data.frame(matrix(ncol = 2))
colnames(tab) <- c("metabolite", "p")
metabolite_long_clean <- metabolite_long %>% 
  filter(sample_type == "leaf") %>% 
  na.omit() %>% 
  filter(metabolite != "17.16 363 273 257" & metabolite != "14.70 284 269 glucoside" & metabolite != "9.82 269 284" & metabolite != "HCH-deltoidin")
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  a <- t.test(value ~ as.factor(as.character(resistance)), data = metabolite_long %>% filter(sample_type == "leaf" & metabolite %in% i & resistance != "trichocarpa - susceptible"), family = "binomial")
  p <- a$p.value
  tab <- rbind(tab, c(i, round(p, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(metabolite_long_clean$metabolite))))
tab <- tab %>% filter(p.adj < 0.05)

metabolite_long %>% 
  filter(sample_type == "leaf" & metabolite == "salirepin" & resistance != "trichocarpa - susceptible") %>%
  ggplot(aes(x = resistance, y = value)) +
  geom_boxplot() +
  scale_y_continuous(name = bquote("Salirepin ("*mu*"g"~g^-1*")")) +
  theme(axis.title.x = element_blank())-> fig.S2a

metabolite_long %>% 
  filter(sample_type == "leaf" & metabolite == "salireposide" & resistance != "trichocarpa - susceptible") %>%
  ggplot(aes(x = resistance, y = value)) +
  scale_y_continuous(name = bquote("Salireposide ("*mu*"g"~g^-1*")")) +
  geom_boxplot() +
  theme(axis.title.x = element_blank())-> fig.S2b

prow <- plot_grid(fig.S2a, fig.S2b, labels = "AUTO", align = "vh")

ggsave("~/Downloads/Figure_S2.pdf", width = 6.5, height = 3.5, units = "in", dpi = 600)
prow
dev.off()

for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long %>% filter(sample_type == "leaf" & metabolite %in% i & resistance != "deltoides")
  a <- wilcox_test(formula = value ~ Infection_2020, data = sub)
  p <- a$p
  tab <- rbind(tab, c(i, round(p, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_leaves_t <- tab %>% filter(p.adj < 0.05)
sig_tab_t <- tab %>% filter(p < 0.05)

# Within only roots
tab <- data.frame(matrix(ncol = 2))
colnames(tab) <- c("metabolite", "p")
metabolite_long_clean <- metabolite_long %>% 
  filter(sample_type == "root") %>% 
  na.omit() %>% 
  filter(metabolite != "17.16 363 273 257" & metabolite != "14.70 284 269 glucoside" & metabolite != "9.82 269 284" & metabolite != "HCH-deltoidin")
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  a <- t.test(value ~ as.factor(as.character(Infection_2020)), data = metabolite_long %>% filter(sample_type == "root" & metabolite %in% i & resistance != "deltoides"), family = "binomial")
  p <- a$p.value
  tab <- rbind(tab, c(i, round(p, 4), round(p.adj, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_roots_t <- tab %>% filter(p < 0.05) # non sig. when p.adjusted

for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long %>% filter(sample_type == "root" & metabolite %in% i & resistance != "deltoides")
  a <- wilcox_test(formula = value ~ Infection_2020, data = sub)
  p <- a$p
  tab <- rbind(tab, c(i, round(p, 4)))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_t <- tab %>% filter(p.adj < 0.05)
sig_tab_t <- tab %>% filter(p < 0.05)

#####
# Are any metabolites correlated with septoria?
####

# leaves
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "rho")
metabolite_long_clean <- metabolite_long %>% filter(sample_type == "leaf") %>%
  na.omit() 
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long_clean %>% filter(metabolite %in% i & resistance != "deltoides" & sample_type == "leaf")
  a <- cor.test(sub$Copy_number_per_ul, sub$value, method = "spearman")
  p <- a$p.value
  rho <- a$estimate
  tab <- rbind(tab, c(i, round(p, 4), rho))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_leaf <- tab %>% filter(p < 0.05) # non sig. when p.adjusted, erythronic acid gamma-lactone and 6-hydroxy-2-cyclohexenone alcohol pos. correlation

# roots
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "rho")
metabolite_long_clean <- metabolite_long %>% filter(sample_type == "root") %>%
  na.omit() 
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long_clean %>% filter(metabolite %in% i & resistance != "deltoides" & sample_type == "root")
  a <- cor.test(sub$Copy_number_per_ul, sub$value, method = "spearman")
  p <- a$p.value
  rho <- a$estimate
  tab <- rbind(tab, c(i, round(p, 4), rho))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_root <- tab %>% filter(p < 0.05) # non sig. when p.adjusted, 12 positive corrlation (11.05 450 217 dehydrosugar and ferulic acid the most) 
#                                          and 2 negative (trichocarpinene and gentisic acid-5-O-glucoside)

# leaves
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "rho")
metabolite_long_clean <- metabolite_long %>% filter(sample_type == "leaf") %>%
  na.omit() 
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long_clean %>% filter(metabolite %in% i & sample_type == "leaf")
  a <- cor.test(sub$Copy_number_per_ul, sub$value, method = "pearson")
  p <- a$p.value
  rho <- a$estimate
  tab <- rbind(tab, c(i, round(p, 4), rho))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_leaf <- tab %>% filter(p < 0.05) # non sig. when p.adjusted, erythronic acid gamma-lactone and 6-hydroxy-2-cyclohexenone alcohol pos. correlation

# roots
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "rho")
metabolite_long_clean <- metabolite_long %>% filter(sample_type == "root") %>%
  na.omit() 
for(i in levels(as.factor(metabolite_long_clean$metabolite))){
  sub <- metabolite_long_clean %>% filter(metabolite %in% i & sample_type == "root")
  a <- cor.test(sub$Copy_number_per_ul, sub$value, method = "pearson")
  p <- a$p.value
  rho <- a$estimate
  tab <- rbind(tab, c(i, round(p, 4), rho))
} # none sig. when adjusted, 
tab$p.adj <- p.adjust(p = tab$p, method = "fdr", n = length(levels(as.factor(tab$metabolite))))
sig_tab_root <- tab %>% filter(p < 0.05) 

### Plots 
# Figure 2
p_data <- metabolite_long %>%
  filter(metabolite %in% sig_tab_leaves_t$metabolite & resistance != "deltoides" & sample_type == "leaf") %>%
  group_by(Infection_2020, metabolite) %>%
  dplyr::summarise(mean = mean(value))

p_data %>% 
  spread(key = Infection_2020, value = mean) %>%
  mutate(y = log2(`1`/`0`)) %>%
  dplyr::mutate(metabolite = fct_reorder(metabolite, as.numeric(y))) %>%
  ggplot(aes(x = reorder(metabolite, y), y = y)) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = 0, ymax = Inf), fill = "#a34d4d", alpha = 0.5) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 0), fill = "gray", alpha = 0.5) +
  geom_bar(stat = "identity")+
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log2 fold change\n(Infected/Uninfected)")) +
  theme(axis.text.y = element_text(size = 8)) +
  coord_flip() -> fig

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_2.pdf", width = 3, height = 4, units = "in", dpi = 600)
fig
dev.off()

p <- fig +
  theme(axis.text = element_text(size = 24), axis.title = element_text(size = 28), axis.text.y = element_text(size = 24)) 

# Figure S1

sub <- metabolites %>% filter(sample_type =="leaf") %>%
                      dplyr::select(-Genotype, -resistance, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul, -species)
x <- colSums(sub) %>% na.omit %>% data.frame %>% rownames()
sub2 <- sub[ , names(sub) %in% x]

ord <- prcomp(sub2)
ord <- merge(ord$x %>% as.data.frame() %>% dplyr::select(PC1, PC2), metabolites %>% filter(sample_type =="leaf") %>% dplyr::select(resistance, Infection_2020), by = 0)

ord %>%
  ggplot(aes(x = PC1, y = PC2, fill = resistance, shape = as.character(Infection_2020))) +
  geom_point(size = 2, color = "black") +
  labs(x = "PC1 (43%)", y = "PC2 (23%)") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), labels = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  stat_ellipse(aes(color = resistance, group = resistance)) +
  scale_color_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), guide = "none") +
  scale_x_continuous(breaks = c(-10000, 0, 10000, 20000)) +
  theme(legend.title = element_blank(), legend.position = "right") +
  guides(fill = guide_legend(override.aes = list(pch = 21, size = 4))) -> fig.S1a

sub <- metabolites %>% filter(sample_type =="root") %>%
                      dplyr::select(-Genotype, -resistance, -SampleID, -Infection_2020, -sample_type, -SampleID, -Copy_number_per_ul, -species)
x <- colSums(sub) %>% na.omit %>% data.frame %>% rownames()
sub2 <- sub[ , names(sub) %in% x]

ord <- prcomp(sub2)
ord <- merge(ord$x %>% as.data.frame() %>% dplyr::select(PC1, PC2), metabolites %>% filter(sample_type =="root") %>% dplyr::select(resistance, Infection_2020), by = 0)

ord %>%
  ggplot(aes(x = PC1, y = PC2, fill = resistance, shape = as.character(Infection_2020))) +
  geom_point(size = 2, color = "black") +
  labs(x = "PC1 (62%)", y = "PC2 (22%)") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), labels = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  stat_ellipse(aes(color = resistance, group = resistance)) +
  scale_color_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), guide = "none") +
  scale_x_continuous(breaks = c(-10000, 0, 10000, 20000)) +
  theme(legend.title = element_blank(), legend.position = "bottom") +
  guides(fill = guide_legend(override.aes = list(pch = 21, size = 4))) -> fig.S1b

fig.S1 <- plot_grid(fig.S1a + theme(legend.position = "none"), fig.S1b + theme(legend.position = "none"), labels = "AUTO", ncol = 2, align = "vh")
leg <- get_legend(fig.S1b + theme(legend.justification = "center", legend.box = "vertical"))
fig.S1 <- plot_grid(fig.S1, leg, ncol = 1, rel_heights = c(1, 0.2))

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_S1.pdf", width = 6.5, height = 3.5, units = "in", dpi = 600)
fig.S1
dev.off()
```

# Microbiome 1 (alpha diversity)
```{r}
# Download data ---------------------------
data_16S <- qza_to_phyloseq_ITS(features = "~/Downloads/table-16S_merge_021121.qza",  
                              taxonomy = "~/Downloads/taxonomy_merged_021121.qza")
metadata_16S <- read_delim("~/Downloads/metadata_16S.txt", delim = "\t")
sep <- read_delim("~/Downloads/Sep_qPCR.txt", delim = "\t")
scoring <- read_excel("~/Downloads/Scoring.xlsx") 
metadata_16S <- merge(metadata_16S, sep %>% dplyr::select(sample_type, SampleID, Copy_number_per_ul), by = c("sample_type", "SampleID"), all.x = T)
metadata_16S <- merge(metadata_16S, metabolites %>% # metabolite df comes from above kernal
                        dplyr::select(-Genotype, -resistance, -Infection_2020, -Copy_number_per_ul, -species) %>%
                        mutate(sample_type = dplyr::recode(sample_type, root = "RE", leaf = "LE")), 
                      by = c("SampleID", "sample_type"), all.x = T)
metadata_16S <- merge(metadata_16S, scoring %>% dplyr::select(SampleID, Infection_2020), by = "SampleID")

sampledata_16S <- sample_data(metadata_16S) #import into phyloseq sample data object
sample_names(sampledata_16S) <- sampledata_16S$`sample-id` # make sure names are the same as the OTU tables
data_16S <- merge_phyloseq(data_16S, sampledata_16S)

data_16S <- subset_taxa(data_16S, Kingdom == "D_0__Bacteria" | Kingdom == "D_0__Archaea")
sum(otu_table(data_16S)) #23509231
data_16S <- subset_taxa(data_16S, Order != "D_3__Chloroplast")
data_16S <- subset_taxa(data_16S, Family != "D_4__Mitochondria")
sum(otu_table(data_16S)) #15277731



data_16S_LE <- prune_samples(sample_sums(data_16S)>=200, data_16S) %>% subset_samples(sample_type == "LE")
data_16S_LW <- prune_samples(sample_sums(data_16S)>=3000, data_16S) %>% subset_samples(sample_type == "LW")
data_16S_RE <- prune_samples(sample_sums(data_16S)>=8000, data_16S) %>% subset_samples(sample_type == "RE")
data_16S_RH <- prune_samples(sample_sums(data_16S)>=10000, data_16S) %>% subset_samples(sample_type == "RH")

data_16S_combined <- merge_phyloseq(data_16S_LE, data_16S_LW, data_16S_RE, data_16S_RH)

data_16S_list <- list(data_16S_LE, data_16S_LW, data_16S_RE, data_16S_RH)
data_16S_combined <- subset_samples(data_16S_combined, SampleID != "S71" & SampleID != "S60" & SampleID != "S62" & SampleID != "S10")




data_ITS <- qza_to_phyloseq_ITS(features = "~/Downloads/table-ITS_merge_021121.qza",  
                              taxonomy = "~/Downloads/taxonomy-ITS_021121.qza")

metadata_ITS <- read_delim("~/Downloads/metadata_ITS.txt", delim = "\t")
sep <- read_delim("~/Downloads/Sep_qPCR.txt", delim = "\t")
scoring <- read_excel("~/Downloads//Scoring.xlsx") 

metadata_ITS <- merge(metadata_ITS, sep %>% dplyr::select(sample_type, SampleID, Copy_number_per_ul), by = c("sample_type", "SampleID"), all.x = T)
metadata_ITS <- merge(metadata_ITS, metabolites %>% # metabolite df comes from above kernal 
                        dplyr::select(-Genotype, -resistance, -Infection_2020, -Copy_number_per_ul, -species) %>%
                        mutate(sample_type = dplyr::recode(sample_type, root = "RE", leaf = "LE")), 
                      by = c("SampleID", "sample_type"), all.x = T)
metadata_ITS <- merge(metadata_ITS, scoring %>% dplyr::select(SampleID, Infection_2020), by = "SampleID")

sampledata_ITS <- sample_data(metadata_ITS) #import into phyloseq sample data object
sample_names(sampledata_ITS) <- sampledata_ITS$`sample-id` # make sure names are the same as the OTU tables
data_ITS <- merge_phyloseq(data_ITS, sampledata_ITS)

data_ITS <- subset_taxa(data_ITS, Kingdom == "k__Fungi")
data_ITS_LE <- prune_samples(sample_sums(data_ITS)>=2000, data_ITS) %>% subset_samples(sample_type == "LE")
data_ITS_LW <- prune_samples(sample_sums(data_ITS)>=4000, data_ITS) %>% subset_samples(sample_type == "LW")
data_ITS_RE <- prune_samples(sample_sums(data_ITS)>=3000, data_ITS) %>% subset_samples(sample_type == "RE")
data_ITS_RH <- prune_samples(sample_sums(data_ITS)>=20000, data_ITS) %>% subset_samples(sample_type == "RH")

data_ITS_combined <- merge_phyloseq(data_ITS_LE, data_ITS_LW, data_ITS_RE, data_ITS_RH)

data_ITS_list <- list(data_ITS_LE, data_ITS_LW, data_ITS_RE, data_ITS_RH)
data_ITS_combined <- subset_samples(data_ITS_combined, SampleID != "S71" & SampleID != "S60" & SampleID != "S62" & SampleID != "S10")

# Calculate Hill numbers ------------
OTU_16S_rare_combined <- read.csv("~/Downloads/data_16S_rare_combined.csv", row.names = "X")
OTU_16S_rare_combined <- otu_table(OTU_16S_rare_combined, taxa_are_rows = T)
sample_names(OTU_16S_rare_combined) <- gsub("[.]", "-", sample_names(OTU_16S_rare_combined))
data_16S_rare_combined <- merge_phyloseq(otu_table(OTU_16S_rare_combined, taxa_are_rows = T), data_16S_combined@tax_table, data_16S_combined@sam_data)
data_16S_rare_combined <- prune_taxa(taxa_sums(data_16S_rare_combined) > 0.5, data_16S_rare_combined)

OTU_ITS_rare_combined <- read.csv("~/Downloads/data_ITS_rare_combined.csv", row.names = "X")
OTU_ITS_rare_combined <- otu_table(OTU_ITS_rare_combined, taxa_are_rows = T)
sample_names(OTU_ITS_rare_combined) <- gsub("[.]", "-", sample_names(OTU_ITS_rare_combined))
data_ITS_rare_combined <- merge_phyloseq(otu_table(OTU_ITS_rare_combined, taxa_are_rows = T), data_ITS_combined@tax_table, data_ITS_combined@sam_data)
data_ITS_rare_combined <- prune_taxa(taxa_sums(data_ITS_rare_combined) > 0.5, data_ITS_rare_combined)

OTU_16S <- t(as(data_16S_rare_combined@otu_table, "matrix"))
OTU_16S <- as.data.frame(OTU_16S)

rich_16S <- hill_taxa(OTU_16S, q = 0)
rich_16S <- merge(data_16S_rare_combined@sam_data, rich_16S, by = 0)

shan_16S <- hill_taxa(OTU_16S, q = 1)
shan_16S <- merge(data_16S_rare_combined@sam_data, shan_16S, by = 0)

simp_16S <- hill_taxa(OTU_16S, q = 2)
simp_16S <- merge(data_16S_rare_combined@sam_data, simp_16S, by = 0)

OTU_ITS <- t(as(data_ITS_rare_combined@otu_table, "matrix"))
OTU_ITS <- as.data.frame(OTU_ITS)

rich_ITS <- hill_taxa(OTU_ITS, q = 0)
rich_ITS <- merge(data_ITS_rare_combined@sam_data, rich_ITS, by = 0)

shan_ITS <- hill_taxa(OTU_ITS, q = 1)
shan_ITS <- merge(data_ITS_rare_combined@sam_data, shan_ITS, by = 0)

simp_ITS <- hill_taxa(OTU_ITS, q = 2)
simp_ITS <- merge(data_ITS_rare_combined@sam_data, simp_ITS, by = 0)

# Diversity by genotype, resitance, and host species and infection --------------
tab_data <- rbind(rich_16S %>% mutate(amplicon = "16S", index = "Richness") %>% dplyr::select(amplicon, index, sample_type, genotype, y, species, Infection_2020, resistance, Copy_number_per_ul),
                  shan_16S %>% mutate(amplicon = "16S", index = "Shannon")  %>% dplyr::select(amplicon, index, sample_type, genotype, y, species, Infection_2020, resistance, Copy_number_per_ul),
                  simp_16S %>% mutate(amplicon = "16S", index = "Simpson")  %>% dplyr::select(amplicon, index, sample_type, genotype, y, species, Infection_2020, resistance, Copy_number_per_ul),
                  rich_ITS %>% mutate(amplicon = "ITS", index = "Richness") %>% dplyr::select(amplicon, index, sample_type, genotype, y, species, Infection_2020, resistance, Copy_number_per_ul),
                  shan_ITS %>% mutate(amplicon = "ITS", index = "Shannon")  %>% dplyr::select(amplicon, index, sample_type, genotype, y, species, Infection_2020, resistance, Copy_number_per_ul),
                  simp_ITS %>% mutate(amplicon = "ITS", index = "Simpson")  %>% dplyr::select(amplicon, index, sample_type, genotype, y, species, Infection_2020, resistance, Copy_number_per_ul))
tab <- data.frame(matrix(ncol = 7, nrow = 0))
colnames(tab) <- c("amplicon","sample_type", "index",  "p.species", "p.genotype", "p.infection", "p.interaction")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k)
      a <- aov(log(y) ~ species, sub)
      p.species <- summary(a)[[1]][["Pr(>F)"]][[1]]
      
      b <- aov(log(y) ~ resistance*as.factor(as.character(Infection_2020)), sub %>% filter(species == "P. trichocarpa"))
      p.genotype <- summary(b)[[1]][["Pr(>F)"]][[1]]
      p.infection <- summary(b)[[1]][["Pr(>F)"]][[2]]
      p.interaction <- summary(b)[[1]][["Pr(>F)"]][[3]]
    
      tab <- rbind(tab, data.frame(i, j, k, p.species, p.genotype, p.infection, p.interaction))
    }
  }
} # decrease in 16S rhizosphere for rich and shan, increase in ITS root endosphere in shan and simp

tab <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab) <- c("amplicon","sample_type", "index", "p.infection")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k)
      a <- t.test(log(y) ~ Infection_2020, sub %>% filter(species == "P. trichocarpa"))
      p.infection <- a$p.value
      tab <- rbind(tab, data.frame(i, j, k, p.infection))
    }
  }
} # decreas

tab <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab) <- c("amplicon","sample_type", "index", "p.infection")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k & species == "P. trichocarpa")
      a <- wilcox_test(formula = y ~ Infection_2020, data = sub )
      p.infection <- a$p
      tab <- rbind(tab, data.frame(i, j, k, p.infection))
    }
  }
} # decreas

tab <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab) <- c("amplicon","sample_type", "index", "p.species")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k)
      a <- wilcox_test(formula = y ~ species, data = sub )
      p.species <- a$p
      tab <- rbind(tab, data.frame(i, j, k, p.species))
    }
  }
} # decreas

tab <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab) <- c("amplicon","sample_type", "index", "p.Genotype")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k)
      a <- aov(log(y) ~ genotype, sub %>% filter(species == "P. trichocarpa"))
      p.Genotype <- summary(a)[[1]][["Pr(>F)"]][[1]]
      tab <- rbind(tab, data.frame(i, j, k, p.Genotype))
    }
  }
} # decreas

tab_data %>%
  filter(amplicon == "16S") %>%
  group_by(sample_type, index, species) %>%
  dplyr::summarise(mean = mean(y))


###
# Any correlations with septoria?
###
tab <- data.frame(matrix(ncol = 7, nrow = 0))
colnames(tab) <- c("amplicon","sample_type", "index", "p.species", "rho", "lwr", "upr")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k & Copy_number_per_ul > 0)
      a <- cor.test(sub$y, sub$Copy_number_per_ul, method = "spearman")
      x <- SpearmanRho(sub$y, sub$Copy_number_per_ul, conf.level = 0.95)
      p.species <- a$p.value
      rho <- a$estimate
      lwr <- x[2]
      upr <- x[3]
      tab <- rbind(tab, data.frame(i, j, k, p.species, rho, lwr, upr))
    }
  }
} # decreas
p_data_alpha_SM <- tab %>%
  mutate(k = dplyr::recode(k, LE = "Leaf\nEndosphere", LW = "Leaf\nSurface", RE = "Root\nEndosphere", RH = "Rhizosphere"),
         j = dplyr::recode(j, Richness = "q = 0", Shannon = "q = 1", Simpson = "q = 2")) %>%
  mutate(k = ordered(k, c("Leaf\nEndosphere", "Leaf\nSurface", "Root\nEndosphere", "Rhizosphere")))

###
# Any correlations with metabolites?
###

tab_data2 <- bind_rows(rich_16S %>% mutate(amplicon = "16S", index = "Richness"),
                       shan_16S %>% mutate(amplicon = "16S", index = "Shannon") ,
                       simp_16S %>% mutate(amplicon = "16S", index = "Simpson") ,
                       rich_ITS %>% mutate(amplicon = "ITS", index = "Richness"),
                       shan_ITS %>% mutate(amplicon = "ITS", index = "Shannon") ,
                       simp_ITS %>% mutate(amplicon = "ITS", index = "Simpson") )

tab_data2_long <- tab_data2 %>% dplyr::select(-resistance, -SampleID, -Copy_number_per_ul, -sample, -genotype, 
                                              -local_sep, -`sample.id`, -`Row.names`, -Infection_2020) %>%
  filter(sample_type %in% c("LE", "RE")) %>%
  gather(key = "metabolite", value = "value", -index, -amplicon, -y, -sample_type, -species)

tab <- data.frame(matrix(ncol = 7, nrow = 0))
colnames(tab) <- c("amplicon", "index", "metabolite", "p", "rho", "lwr", "upr")
tab_data2_long_clean <- tab_data2_long %>% na.omit() %>% filter(sample_type == "LE")
  #filter(metabolite != "17.16 363 273 257" & metabolite != "14.70 284 269 glucoside" & metabolite != "9.82 269 284" & metabolite != "HCH-deltoidin")
for(i in levels(as.factor(as.character(tab_data2_long_clean$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data2_long_clean$index)))){
    for(k in levels(as.factor(as.character(tab_data2_long_clean$metabolite)))){
      sub <- subset(tab_data2_long_clean, amplicon %in% i & index %in% j & metabolite %in% k & value > 0)
      a <- cor.test(sub$y, sub$value, method = "spearman")
      x <- SpearmanRho(sub$y, sub$value, conf.level = 0.95)
      p <- a$p.value
      rho <- a$estimate
      lwr <- x[2]
      upr <- x[3]
      tab <- rbind(tab, data.frame(i, j, k, p, rho, lwr, upr))
    }
  }
} # 
tab_rich <- tab %>% filter(j == "Richness")
tab_rich$p.adj <- p.adjust(tab_rich$p, method = "fdr")
tab_shan <- tab %>% filter(j == "Shannon")
tab_shan$p.adj <- p.adjust(tab_shan$p, method = "fdr")
tab_simp <- tab %>% filter(j == "Simpson")
tab_simp$p.adj <- p.adjust(tab_simp$p, method = "fdr")
tab_all <- rbind(tab_rich, tab_shan, tab_simp)
sig_tab_leaf <- tab_all %>% filter(p.adj < 0.05)

tab <- data.frame(matrix(ncol = 7, nrow = 0))
colnames(tab) <- c("amplicon", "index", "metabolite", "p", "rho", "lwr", "upr")
tab_data2_long_clean <- tab_data2_long %>% na.omit() %>% filter(sample_type == "RE")
  #filter(metabolite != "17.16 363 273 257" & metabolite != "14.70 284 269 glucoside" & metabolite != "9.82 269 284" & metabolite != "HCH-deltoidin")
for(i in levels(as.factor(as.character(tab_data2_long_clean$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data2_long_clean$index)))){
    for(k in levels(as.factor(as.character(tab_data2_long_clean$metabolite)))){
      sub <- subset(tab_data2_long_clean, amplicon %in% i & index %in% j & metabolite %in% k & value > 0)
      a <- cor.test(sub$y, sub$value, method = "spearman")
      x <- SpearmanRho(sub$y, sub$value, conf.level = 0.95)
      p <- a$p.value
      rho <- a$estimate
      lwr <- x[2]
      upr <- x[3]
      tab <- rbind(tab, data.frame(i, j, k, p, rho, lwr, upr))
    }
  }
} # 
tab_rich <- tab %>% filter(j == "Richness")
tab_rich$p.adj <- p.adjust(tab_rich$p, method = "fdr")
tab_shan <- tab %>% filter(j == "Shannon")
tab_shan$p.adj <- p.adjust(tab_shan$p, method = "fdr")
tab_simp <- tab %>% filter(j == "Simpson")
tab_simp$p.adj <- p.adjust(tab_simp$p, method = "fdr")
tab_all <- rbind(tab_rich, tab_shan, tab_simp)
sig_tab_root <- tab_all %>% filter(p.adj < 0.05)


# plot ------------
p_data_alpha_SM %>%
  filter(i == "16S") %>%
  ggplot(aes(x = k, y = rho, fill = j, shape = j)) +
  geom_errorbar(aes(ymin = lwr, ymax = upr), position = position_dodge(0.9), width = 0.2) +
  geom_point(col = "black", stat = "identity", position = position_dodge(0.9), size = 3) +
  scale_shape_manual(values = c(21,23,24)) +
  scale_x_discrete(name = NULL) +
  geom_hline(yintercept = 0, linetype = "dashed") +
  theme(axis.text.x = element_text(size = 9)) -> fig.3a
p_data_alpha_SM %>%
  filter(i == "ITS") %>%
  ggplot(aes(x = k, y = rho, fill = j, shape = j)) +
  geom_errorbar(aes(ymin = lwr, ymax = upr), position = position_dodge(0.9), width = 0.2) +
  geom_point(col = "black", stat = "identity", position = position_dodge(0.9), size = 3) +
  scale_shape_manual(values = c(21,23,24)) +
  scale_x_discrete(name = NULL) +
  geom_hline(yintercept = 0, linetype = "dashed") +
  theme(axis.text.x = element_text(size = 9)) -> fig.3b

fig.3 <- plot_grid(fig.3a, fig.3b, labels = "AUTO", ncol = 1, align = "vh")
leg <- get_legend(fig.3b + theme(legend.position = "bottom", legend.title = element_blank(), legend.justification = "center"))
fig.3 <- plot_grid(fig.3, leg, ncol = 1, rel_heights = c(1, 0.1))

ggsave("~/Downloads/Figure3.pdf", width = 3.25, height = 6, units = "in", dpi = 600)
fig.3
dev.off()

sig_tab_leaf %>% 
  filter(j == "Richness") %>%
  mutate(k = gsub(x = k, pattern = ".", replacement = " ", fixed = T)) %>%
  mutate(k = dplyr::recode(k, `trans 3 O caffeoylquinic acid` = "trans-3-O-caffeoylquinic acid", 
                           `X16 15 297 361 209` = "19.19 min m/z 16 15 297 361 209",
                           `X2 5 dihydroxybenzoic acid` = "2,5-dihydroxybenzoic acid")) %>%
  dplyr::mutate(k = fct_reorder(k, as.numeric(rho))) %>%
  ggplot(aes(y = k, x = rho)) +
  geom_errorbarh(aes(xmin = lwr, xmax = upr), height = 0.2) +
  geom_point(stat = "identity", size = 2, pch = 21, col = "black", fill = "gray") +
  scale_y_discrete(name = NULL) +
  geom_vline(xintercept = 0, linetype = "dashed") -> fig.4

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure4.pdf", width = 3.25, height = 4, units = "in", dpi = 600)
fig.4
dev.off()

plot_data <- rbind(rich_16S %>% mutate(amplicon = "16S") %>% dplyr::select(y, sample_type, amplicon, genotype, resistance, Infection_2020), 
                   rich_ITS %>% mutate(amplicon = "ITS") %>% dplyr::select(y, sample_type, amplicon, genotype, resistance, Infection_2020))

plot_data %>% na.omit() %>%
  dplyr::mutate(genotype = ordered(genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(sample_type, amplicon, genotype, resistance) %>%
  dplyr::summarise(mean = mean(y), se = sd(y)/sqrt(n())) %>%
  ggplot(aes(x = genotype, y = mean, fill = resistance)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  geom_jitter(data = plot_data, aes(x = genotype, y = y, fill = resistance, shape = as.factor(as.character(Infection_2020))), inherit.aes = F, alpha = 0.5) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  geom_errorbar(aes(ymin = mean -se, ymax = mean + se), width = 0.2) +
  facet_grid(amplicon~sample_type, scales = "free", 
             labeller = labeller(sample_type = c("LE" = "Leaf Endosphere", "LW" = "Leaf Surface",  "RE" = "Root Endosphere", "RH" = "Rhizosphere"),
                                 amplicon = c("16S" = "Archea & Bacteria", "ITS" = "Fungi"))) +
  scale_y_continuous(name = "Hill Number (q = 0)") +
  panel_border() +
  theme(axis.text.x = element_text(angle = 45, size = 10, hjust = 1), legend.title = element_blank(), legend.justification = "center",
        legend.position = "bottom", legend.text = element_text(size = 12), axis.title.x = element_blank(), legend.box = "vertical") -> fig.3

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure3.pdf", width = 7.5, height = 5, units = "in", dpi = 600)
fig.3
dev.off()

plot_data <- rbind(shan_16S %>% mutate(amplicon = "16S") %>% dplyr::select(y, sample_type, amplicon, genotype, resistance, Infection_2020), 
                   shan_ITS %>% mutate(amplicon = "ITS") %>% dplyr::select(y, sample_type, amplicon, genotype, resistance, Infection_2020))
plot_data %>% na.omit() %>%
  dplyr::mutate(genotype = ordered(genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(sample_type, amplicon, genotype, resistance) %>%
  dplyr::summarise(mean = mean(y), se = sd(y)/sqrt(n())) %>%
  ggplot(aes(x = genotype, y = mean, fill = resistance)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  geom_jitter(data = plot_data, aes(x = genotype, y = y, fill = resistance, shape = as.factor(as.character(Infection_2020))), inherit.aes = F, alpha = 0.5) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  geom_errorbar(aes(ymin = mean -se, ymax = mean + se), width = 0.2) +
  facet_grid(amplicon~sample_type, scales = "free", 
             labeller = labeller(sample_type = c("LE" = "Leaf Endosphere", "LW" = "Leaf Surface",  "RE" = "Root Endosphere", "RH" = "Rhizosphere"),
                                 amplicon = c("16S" = "Archea & Bacteria", "ITS" = "Fungi"))) +
  scale_y_continuous(name = "Hill number (q = 1)") +
  panel_border() +
  theme(axis.text.x = element_text(angle = 45, size = 8, hjust = 1), legend.title = element_blank(),
        legend.position = c(0.01, 0.82), legend.text = element_text(size = 8), axis.title.x = element_blank(), 
        legend.spacing.y = unit(0.07, "cm")) -> fig.S2a

plot_data <- rbind(simp_16S %>% mutate(amplicon = "16S") %>% dplyr::select(y, sample_type, amplicon, genotype, resistance, Infection_2020), 
                   simp_ITS %>% mutate(amplicon = "ITS") %>% dplyr::select(y, sample_type, amplicon, genotype, resistance, Infection_2020))
plot_data %>% na.omit() %>%
  dplyr::mutate(genotype = ordered(genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(sample_type, amplicon, genotype, resistance) %>%
  dplyr::summarise(mean = mean(y), se = sd(y)/sqrt(n())) %>%
  ggplot(aes(x = genotype, y = mean, fill = resistance)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  geom_jitter(data = plot_data, aes(x = genotype, y = y, fill = resistance, shape = as.factor(as.character(Infection_2020))), inherit.aes = F, alpha = 0.5) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  geom_errorbar(aes(ymin = mean -se, ymax = mean + se), width = 0.2) +
  facet_grid(amplicon~sample_type, scales = "free", 
             labeller = labeller(sample_type = c("LE" = "Leaf Endosphere", "LW" = "Leaf Surface",  "RE" = "Root Endosphere", "RH" = "Rhizosphere"),
                                 amplicon = c("16S" = "Archea & Bacteria", "ITS" = "Fungi"))) +
  scale_y_continuous(name = "Hill number (q = 2)") +
  panel_border() +
  theme(axis.text.x = element_text(angle = 45, size = 8, hjust = 1), legend.title = element_blank(),
        legend.position = "right", legend.text = element_text(size = 8), axis.title.x = element_blank()) -> fig.S2b

fig.S2 <- plot_grid(fig.S2a, fig.S2b + theme(legend.position = "none"), ncol = 1, align = "vh", labels = "AUTO")

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_S2.pdf", width = 7.5, height = 8, units = "in", dpi = 600)
fig.S2
dev.off()

# Is diveristy impacted by metabolites? -----------------------
tab_16S_rich <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab_16S_rich) <- c("name", "p.adj", "p", "rho")
for(i in 11:161){
  sub <- subset(rich_16S, sample_type == "RE")
  a <- cor.test(sub$y, sub[,i], method = "spearman")
  p.adj <- p.adjust(a$p.value, n = 150, method = "fdr")
  p <- a$p.value
  rho <- a$estimate
  tab_16S_rich <- rbind(tab_16S_rich, data.frame(i, p.adj, p, rho))
} # none sig.
tab_16S_shan <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab_16S_shan) <- c("name", "p.adj", "p", "rho")
for(i in 11:161){
  sub <- subset(shan_16S, sample_type == "RE")
  a <- cor.test(sub$y, sub[,i], method = "spearman")
  p.adj <- p.adjust(a$p.value, n = 150, method = "fdr")
  p <- a$p.value
  rho <- a$estimate
  tab_16S_shan <- rbind(tab_16S_shan, data.frame(i, p.adj, p, rho))
} # 138 is sig. negative
tab_16S_simp <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab_16S_simp) <- c("name", "p.adj", "p", "rho")
for(i in 11:161){
  sub <- subset(simp_16S, sample_type == "RE")
  a <- cor.test(sub$y, sub[,i], method = "spearman")
  p.adj <- p.adjust(a$p.value, n = 150, method = "fdr")
  p <- a$p.value
  rho <- a$estimate
  tab_16S_simp <- rbind(tab_16S_simp, data.frame(i, p.adj, p, rho))
} # none sig.

tab_ITS_rich <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab_ITS_rich) <- c("name", "p.adj", "p", "rho")
for(i in 10:160){
  sub <- subset(rich_ITS, sample_type == "RE")
  a <- cor.test(sub$y, sub[,i], method = "spearman")
  p.adj <- p.adjust(a$p.value, n = 150, method = "fdr")
  p <- a$p.value
  rho <- a$estimate
  tab_ITS_rich <- rbind(tab_ITS_rich, data.frame(i, p.adj, p, rho))
} # none sig.
tab_ITS_shan <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab_ITS_shan) <- c("name", "p.adj", "p", "rho")
for(i in 10:160){
  sub <- subset(shan_ITS, sample_type == "RE")
  a <- cor.test(sub$y, sub[,i], method = "spearman")
  p.adj <- p.adjust(a$p.value, n = 150, method = "fdr")
  p <- a$p.value
  rho <- a$estimate
  tab_ITS_shan <- rbind(tab_ITS_shan, data.frame(i, p.adj, p, rho))
} # none sig.
tab_ITS_simp <- data.frame(matrix(ncol = 4, nrow = 0))
colnames(tab_ITS_simp) <- c("name", "p.adj", "p", "rho")
for(i in 10:160){
  sub <- subset(simp_ITS, sample_type == "RE")
  a <- cor.test(sub$y, sub[,i], method = "spearman")
  p.adj <- p.adjust(a$p.value, n = 150, method = "fdr")
  p <- a$p.value
  rho <- a$estimate
  tab_ITS_simp <- rbind(tab_ITS_simp, data.frame(i, p.adj, p, rho))
} # none sig.

colnames(shan_16S)[138]

# Plot --------
shan_16S %>%
  filter(sample_type == "RE") %>%
  ggplot(aes(y = y, x = X19.19.451.395.587.602.512, fill = resistance, shape = as.character(Infection_2020))) +
  geom_point() +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  scale_x_continuous(name = bquote("19.19 min m/z 451 395 587 602 512 ("*mu*"g"~g^-1*")")) +
  scale_y_continuous(name = "Shannon diversity (Hill number)") +
  theme(legend.position = "right", legend.title = element_blank()) +
  guides(fill = guide_legend(override.aes = list(pch = 21, size  =2))) -> fig.S3

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_S3.pdf", width = 6.5, height = 4, units = "in", dpi = 600)
fig.S3
dev.off()

# Is diveristy impacted by sep? -----------------------

tab_data <- rbind(rich_16S %>% mutate(amplicon = "16S", index = "Richness") %>% dplyr::select(amplicon, index, sample_type, Copy_number_per_ul, y, resistance, Infection_2020),
                  shan_16S %>% mutate(amplicon = "16S", index = "Shannon")  %>% dplyr::select(amplicon, index, sample_type, Copy_number_per_ul, y, resistance, Infection_2020),
                  simp_16S %>% mutate(amplicon = "16S", index = "Simpson")  %>% dplyr::select(amplicon, index, sample_type, Copy_number_per_ul, y, resistance, Infection_2020),
                  rich_ITS %>% mutate(amplicon = "ITS", index = "Richness") %>% dplyr::select(amplicon, index, sample_type, Copy_number_per_ul, y, resistance, Infection_2020),
                  shan_ITS %>% mutate(amplicon = "ITS", index = "Shannon")  %>% dplyr::select(amplicon, index, sample_type, Copy_number_per_ul, y, resistance, Infection_2020),
                  simp_ITS %>% mutate(amplicon = "ITS", index = "Simpson")  %>% dplyr::select(amplicon, index, sample_type, Copy_number_per_ul, y, resistance, Infection_2020))
tab <- data.frame(matrix(ncol = 5, nrow = 0))
colnames(tab) <- c("amplicon", "index", "sample_type", "p", "rho")
for(i in levels(as.factor(as.character(tab_data$amplicon)))){
  for(j in levels(as.factor(as.character(tab_data$index)))){
    for(k in levels(as.factor(as.character(tab_data$sample_type)))){
      sub <- subset(tab_data, amplicon %in% i & index %in% j & sample_type %in% k)
      a <- cor.test(sub$y, sub$Copy_number_per_ul, method = "spearman")
      p <- a$p.value
      rho <- a$estimate
      tab <- rbind(tab, data.frame(i, j, k, p, rho))
    }
  }
} # decrease in 16S rhizosphere for rich and shan, increase in ITS root endosphere in shan and simp

# Plot --------------------
tab_data %>% 
  filter(sample_type == "RE") %>%
  ggplot(aes(x = Copy_number_per_ul*30/50, y = y, fill = resistance, shape = as.character(Infection_2020))) +
  geom_point() +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  scale_x_continuous(trans = "log10", name = bquote(italic("S. musiva")~"abundance (copy"~mg^-1*")")) +
  scale_y_continuous("Hill Number") +
  facet_grid(amplicon~index, scale = "free", labeller = labeller(amplicon = c("16S" = "Archea & Bacteria", "ITS" = "Fungi"))) +
  stat_smooth(aes(alpha = interaction(amplicon, index), group = interaction(amplicon, index)), 
              color = "blue", size = 1, geom = "line", method = "lm") +
  scale_alpha_manual(values = c(0,0,0,1,0,1), guide = F) +
  panel_border() +
  geom_text(data = data.frame(index = c("Shannon", "Shannon", "Simpson", "Simpson"),
                              amplicon = rep("ITS", 4),
                              label = c("p = 0.017", "rho = 0.298", "p = 0.004", "rho = 0.354"),
                              x = rep(0.1, 4),
                              y = c(70, 60, 70, 60)),
            aes(x = x, y = y, label = label), inherit.aes = F, hjust = 0) +
  theme(legend.position = c(0.68, 0.83), legend.title = element_blank(), legend.text = element_text(size = 8.5), 
        legend.spacing.y = unit(0.06, 'cm')) +
  guides(fill = guide_legend(override.aes = list(pch = 21, size  =2))) -> figS4a

tab_data %>% 
  filter(sample_type == "RH") %>%
  ggplot(aes(x = Copy_number_per_ul*50/250, y = y, fill = resistance, shape = as.character(Infection_2020))) +
  geom_point() +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  scale_x_continuous(trans = "log10", name = bquote(italic("S. musiva")~"abundance (copy"~mg^-1*")")) +
  scale_y_continuous("Hill Number") +
  facet_grid(amplicon~index, scale = "free", labeller = labeller(amplicon = c("16S" = "Archea & Bacteria", "ITS" = "Fungi"))) +
  stat_smooth(aes(alpha = interaction(amplicon, index), group = interaction(amplicon, index)), 
              color = "blue", size = 1, geom = "line", method = "lm") +
  scale_alpha_manual(values = c(1,0,1,0,0,0)) +
  panel_border() +
  geom_text(data = data.frame(index = c("Richness", "Richness", "Shannon", "Shannon"),
                              amplicon = rep("16S", 4),
                              label = c("p = 0.010", "rho = -0.301", "p = 0.025", "rho = -0.265"),
                              x = rep(0.1, 4),
                              y = c(700, 300, 2800, 2400)),
            aes(x = x, y = y, label = label), inherit.aes = F, hjust = 0)-> figS4b

fig.S4 <- plot_grid(figS4a, figS4b, ncol = 1, align = "vh", labels = "AUTO")

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_S4.pdf", width = 6.5, height = 8, units = "in", dpi = 600)
fig.S4
dev.off()

```

# Correlation between ITS and qPCR Septoria
```{r}
data_ITS_combined %>%
  microbiome::transform("compositional") %>%
  subset_taxa(Genus == "g__Septoria" | Genus == "g__Sphaerulina") %>%
  tax_glom("Genus") %>%
  psmelt() %>%
  filter(sample_type == "LE" | sample_type == "RE") %>%
  dplyr::select(SampleID, sample_type, species, Copy_number_per_ul, Abundance, Genus) %>%
  spread(key = "Genus", value = "Abundance") %>%
  mutate(total = g__Sphaerulina + g__Septoria) -> data
  
cor.test(data$total, data$Copy_number_per_ul) # not sig
cor.test(data$g__Septoria, data$Copy_number_per_ul) # not sig
cor.test(data$g__Sphaerulina, data$Copy_number_per_ul) # not sig


data_LE <- data %>% filter(sample_type == "LE")
cor.test(data_LE$total, data_LE$Copy_number_per_ul) # not sig
cor.test(data_LE$g__Septoria, data_LE$Copy_number_per_ul) # not sig
cor.test(data_LE$g__Sphaerulina, data_LE$Copy_number_per_ul) # not sig

data_ITS_combined %>%
  microbiome::transform("compositional") %>%
  subset_taxa(Genus == "g__Septoria" | Genus == "g__Sphaerulina") %>%
  tax_glom("Genus") %>%
  subset_samples(sample_type == "LE" | sample_type == "RE") -> sub

#sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "stearic.acid", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res

```

# Microbiome 2 (beta diversity)
```{r}
data_16S_tss <- microbiome::transform(data_16S_combined, transform = "compositional")
Acinetobacter <- subset_taxa(data_16S_tss, Genus == "D_5__Acinetobacter")
Acinetobacter_glom <- tax_glom(Acinetobacter, "Genus")
Acinetobacter_glom %>%
  psmelt %>%
  group_by(sample_type, species) %>%
  summarise(mean = mean(Abundance*100), se = sd(Abundance*100)/sqrt(n())) %>%
  ggplot(aes(x = sample_type, y = mean, fill = species)) +
  geom_bar(stat = "identity", position = position_dodge(0.9)) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), position = position_dodge(0.9), width = 0.2) +
  scale_x_discrete(name = NULL, labels = c("Leaf\nEndosphere", "Leaf\nSurface", "Root\nEndosphere", "Rhizosphere")) +
  scale_y_continuous("Relative Abundance (%)")-> p


data <- data.frame(Month = c(rep("June", 4), rep("July", 4), rep("August", 4), rep("September", 4)),
                   Treatment = c("SCE", "STS", "GS", "Control", "SCE", "STS", "GS", "Control","SCE", "STS", "GS", "Control", "SCE", "STS", "GS", "Control"),
                   mean = c(11, 4.75, 3.5, 4.5, 11.5, 4.75, 4, 5.25, 13.25, 4.5, 5, 5.75, 16, 5.5, 7.25, 6.25),
                   error = c(3, 1.25, 0.5, 1.5, 2, 1.25, 0.5, 1.25, 2, 1.5, 0.75, 0.75, 0.75, 1, 0.75, 0.3))
data$Treatment <- ordered(data$Treatment, c("Control", "GS", "STS", "SCE"))
data %>%
  ggplot(aes(x = Month, fill = Treatment, y = mean)) +
  geom_bar(stat = "identity", position = position_dodge(0.9), col = "black", lwd = 0.25) +
  geom_errorbar(aes(ymin = mean - error, ymax = mean + error), position = position_dodge(0.9), width = 0.2) +
  scale_fill_manual(values = c("grey", "#28334AFF", "#FBDE44FF", "#F65058FF"), 
                    labels = c("Control", "Group Selection", "Single Tree Selection", "Structural Complexity Enhancement")) +
  labs(y = "Richness")  +
  theme(axis.title = element_text(size = 14), axis.text = element_text(size = 12), legend.text = element_text(size = 14)) -> p

ggsave("~/Desktop/DoveKeeton2015fig.png", width = 8, height = 4, units = "in", dpi = 600)
p
dev.off()

### Overall --------------------
# 16S ----------------------
data_16S_tss <- microbiome::transform(data_16S_combined, transform = "compositional")
dist <- distance(data_16S_tss, "bray")
df_16S <- data.frame(sample_data(data_16S_tss))
set.seed(01221990)
ad <- vegan::adonis(dist ~ sample_type*resistance*Infection_2020, df_16S) # large differences by sample type

for(i in levels(as.factor(data_16S_tss@sam_data$sample_type))){
  sub <- subset_samples(data_16S_tss, sample_type %in% i)
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ species, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # species effect in everything but RE

for(i in levels(as.factor(data_16S_tss@sam_data$sample_type))){
  sub <- subset_samples(data_16S_tss, sample_type %in% i & species == "P. trichocarpa")
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ resistance, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # only thing is an interaction for rhizosphere

sub <- subset_samples(data_16S_tss, sample_type == "RH" & resistance == "trichocarpa - susceptible")
dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Infection_2020, df_16S) 
ad # infection effect

sub <- subset_samples(data_16S_tss, sample_type == "RH" & resistance == "trichocarpa - resistant")
dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Infection_2020, df_16S) 
ad # no infection effect

# ITS--------------

data_ITS_tss <- microbiome::transform(data_ITS_combined, transform = "compositional")
dist <- distance(data_ITS_tss, "bray")
df_ITS <- data.frame(sample_data(data_ITS_tss))
set.seed(01221990)
ad <- vegan::adonis(dist ~ sample_type*genotype*Infection_2020, df_ITS) # large differences by sample type, small host species effect

for(i in levels(as.factor(data_ITS_tss@sam_data$sample_type))){
  sub <- subset_samples(data_ITS_tss, sample_type %in% i)
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ species, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # species effect in everything but RE

for(i in levels(as.factor(data_ITS_tss@sam_data$sample_type))){
  sub <- subset_samples(data_ITS_tss, sample_type %in% i & species == "P. trichocarpa")
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ Infection_2020, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # only thing is an interaction for rhizosphere

for(i in levels(as.factor(data_ITS_tss@sam_data$sample_type))){
  sub <- subset_samples(data_ITS_tss, sample_type %in% i & species == "P. trichocarpa")
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ resistance, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # only thing is an interaction for rhizosphere

sub <- subset_samples(data_ITS_tss, sample_type == "LE" & resistance == "trichocarpa - susceptible")
dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Infection_2020, df_16S) 
ad # no infection effect

sub <- subset_samples(data_ITS_tss, sample_type == "LE" & resistance == "trichocarpa - resistant")
dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Infection_2020, df_16S) 
ad # no infection effect

sub <- subset_samples(data_ITS_tss, sample_type == "LW" & resistance == "trichocarpa - susceptible")
dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Infection_2020, df_16S) 
ad # no infection effect

sub <- subset_samples(data_ITS_tss, sample_type == "LW" & resistance == "trichocarpa - resistant")
dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Infection_2020, df_16S) 
ad # infection effect

# What about the effect of S. musiva abundance?

for(i in levels(as.factor(data_16S_tss@sam_data$sample_type))){
  sub <- subset_samples(data_16S_tss, sample_type %in% i & Copy_number_per_ul > 0)
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ Copy_number_per_ul, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # species effect in everything but RE

for(i in levels(as.factor(data_ITS_tss@sam_data$sample_type))){
  sub <- subset_samples(data_ITS_tss, sample_type %in% i & Copy_number_per_ul > 0)
  dist <- distance(sub, "bray")
  df_16S <- data.frame(sample_data(sub))
  set.seed(01221990)
  ad <- vegan::adonis(dist ~ Copy_number_per_ul, df_16S) # large differences by sample type
  print(i)
  print(ad)
} # species effect in everything but RE

# Differentially abundant infection vs. non infection -------------
# 16S -------------
Phylum_16S <- data_16S_combined %>% tax_glom("Phylum") 
Class_16S <- data_16S_combined %>% tax_glom("Class") 
Family_16S <- data_16S_combined %>% tax_glom("Family") 
Genus_16S <- data_16S_combined %>% tax_glom("Genus") 

Phylum_ITS <- data_ITS_combined %>% tax_glom("Phylum") 
Class_ITS <- data_ITS_combined %>% tax_glom("Class") 
Family_ITS <- data_ITS_combined %>% tax_glom("Family") 
Genus_ITS <- data_ITS_combined %>% tax_glom("Genus") 

tab <- data.frame(matrix(ncol = 5, nrow = 0))
#colnames(tab) <- c("OTU", "q", "b", "se", "sample_type")
for(i in levels(as.factor(Phylum_16S@sam_data$sample_type))){
  sub <- subset_samples(Phylum_16S, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_phylum <- merge(sig_tab, Phylum_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
#colnames(tab) <- c("OTU", "q", "b", "se", "sample_type")
for(i in levels(as.factor(Class_16S@sam_data$sample_type))){
  sub <- subset_samples(Class_16S, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Class <- merge(sig_tab, Class_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Family_16S@sam_data$sample_type))){
  sub <- subset_samples(Family_16S, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Fam <- merge(sig_tab, Family_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Genus_16S@sam_data$sample_type))){
  sub <- subset_samples(Genus_16S, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Genus <- merge(sig_tab, Genus_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(data_16S_combined@sam_data$sample_type))){
  sub <- subset_samples(data_16S_combined, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_OTU <- merge(sig_tab, data_16S_combined@tax_table, by.x = "OTU", by.y = 0, all.x = T)

# ITS --------------------------------------
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Phylum_ITS@sam_data$sample_type))){
  sub <- subset_samples(Phylum_ITS, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_phylum_ITS <- merge(sig_tab, Phylum_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Class_ITS@sam_data$sample_type))){
  sub <- subset_samples(Class_ITS, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Class_ITS <- merge(sig_tab, Class_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Family_ITS@sam_data$sample_type))){
  sub <- subset_samples(Family_ITS, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Fam_ITS <- merge(sig_tab, Family_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Genus_ITS@sam_data$sample_type))){
  sub <- subset_samples(Genus_ITS, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Genus_ITS <- merge(sig_tab, Genus_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(data_ITS_combined@sam_data$sample_type))){
  sub <- subset_samples(data_ITS_combined, sample_type %in% i & species == "P. trichocarpa")
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Infection_2020", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_OTU_ITS <- merge(sig_tab, data_ITS_combined@tax_table, by.x = "OTU", by.y = 0, all.x = T)

# Differentially abundant with S.musiva abundance -------------
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Phylum_16S@sam_data$sample_type))){
  sub <- subset_samples(Phylum_16S, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_phylum_16S_SM <- merge(sig_tab, Phylum_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Class_16S@sam_data$sample_type))){
  sub <- subset_samples(Class_16S, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Class_16S_SM <- merge(sig_tab, Class_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Family_16S@sam_data$sample_type))){
  sub <- subset_samples(Family_16S, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Fam_16S_SM <- merge(sig_tab, Family_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Genus_16S@sam_data$sample_type))){
  sub <- subset_samples(Genus_16S, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Genus_16S_SM <- merge(sig_tab, Genus_16S@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(data_16S_combined@sam_data$sample_type))){
  sub <- subset_samples(data_16S_combined, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_OTU_16S_SM <- merge(sig_tab, data_16S_combined@tax_table, by.x = "OTU", by.y = 0, all.x = T)

# ITS --------------------------------------
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Phylum_ITS@sam_data$sample_type))){
  sub <- subset_samples(Phylum_ITS, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_phylum_ITS_SM <- merge(sig_tab, Phylum_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Class_ITS@sam_data$sample_type))){
  sub <- subset_samples(Class_ITS, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Class_ITS_SM <- merge(sig_tab, Class_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Family_ITS@sam_data$sample_type))){
  sub <- subset_samples(Family_ITS, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Fam_ITS_SM <- merge(sig_tab, Family_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(Genus_ITS@sam_data$sample_type))){
  sub <- subset_samples(Genus_ITS, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_Genus_ITS_SM <- merge(sig_tab, Genus_ITS@tax_table, by.x = "OTU", by.y = 0, all.x = T)

tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in levels(as.factor(data_ITS_combined@sam_data$sample_type))){
  sub <- subset_samples(data_ITS_combined, sample_type %in% i)
  sub = prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$sample_type <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab <- bind_rows(tab, tab2)
}
sig_tab <- tab %>% filter(q < 0.05)
sig_tab_OTU_ITS_SM <- merge(sig_tab, data_ITS_combined@tax_table, by.x = "OTU", by.y = 0, all.x = T)

sig_tab_OTU_16S_SM$pos.neg <- "pos"
sig_tab_OTU_16S_SM$pos.neg[sig_tab_OTU_16S_SM$b < 0] <- "neg"
sig_tab_OTU_16S_SM %>%
  group_by(sample_type, pos.neg) %>%
  tally

sig_tab_OTU_ITS_SM$pos.neg <- "pos"
sig_tab_OTU_ITS_SM$pos.neg[sig_tab_OTU_ITS_SM$b < 0] <- "neg"
sig_tab_OTU_ITS_SM %>%
  group_by(sample_type, pos.neg, Phylum) %>%
  tally

# Plots

plot_data <- rbind(sig_tab_phylum_16S_SM, sig_tab_phylum_ITS_SM)
plot_data$Phylum <- sapply(strsplit(as.character(plot_data$Phylum), "__"), `[`, 2)
plot_data$int <- interaction(plot_data$sample_type, plot_data$Phylum)
tex <- data.frame(label = c("Leaf Endosphere", "Root Endosphere"),
                  x = rep(-0.01,2),
                  y = c(10, 7))
plot_data %>% 
  dplyr::mutate(int = fct_reorder(int, b)) %>%
  dplyr::mutate(int = fct_reorder(int, sample_type, .desc = T)) %>%
  ggplot(aes(x = b, y = int, fill = Kingdom)) +
  geom_bar(stat = "identity") +
  geom_errorbarh(aes(xmin = b - se$Copy_number_per_ul, xmax = b + se$Copy_number_per_ul), height = 0.2) +
  geom_hline(yintercept = c(1.5, 8.5, 17.5), linetype = "dashed") +
  geom_vline(xintercept = 0)  +
  scale_y_discrete(labels = c("Aphelidiomycota", "Elusimicrobia", "Deinococcus-Thermus", "Nitrospirae", "WS2", 
                              "Patescibacteria", "Ascomycota", "Fibrobacteres", "Deinococcus-Thermus", "Proteobacteria", 
                              "Actinobacteria", "BRC1", "Bacteroidetes", "Latescibacteria", "Nitrospirae", "Thaumarchaeota", 
                              "Chytridiomycota", "Deinococcus-Thermus", "Chloroflexi", "Basidiomycota"),
                   name = NULL) +
  scale_fill_manual(values = c("#920001", "#004949", "#693A91"), labels = c("Archaea", "Bacteria", "Fungi")) +
  scale_x_continuous(name = bquote("Log Fold Change per"~italic(S.)~italic(musiva)*" Copy")) +
  geom_text(data = tex, aes(label = label, x = x, y = y), inherit.aes = F, size = 5) +
  theme(legend.position = "right")-> fig.5a


tex <- data.frame(label = c("Leaf\nEndosphere", "Root\nEndosphere"),
                  x = rep(-0.013,2),
                  y = c(9.5, 6))
plot_data %>% 
  filter(sample_type == "LE" | sample_type == "RE") %>%
  dplyr::mutate(int = fct_reorder(int, b)) %>%
  dplyr::mutate(int = fct_reorder(int, sample_type, .desc = T)) %>%
  ggplot(aes(x = b, y = int, fill = Kingdom)) +
  geom_bar(stat = "identity") +
  geom_errorbarh(aes(xmin = b - se$Copy_number_per_ul, xmax = b + se$Copy_number_per_ul), height = 0.2) +
  geom_vline(xintercept = 0)  +
  scale_y_discrete(labels = c("Elusimicrobia", "Deinococcus-Thermus", "Nitrospirae", 
                              "WS2", "Patescibacteria", "Ascomycota", "Fibrobacteres", 
                              "Deinococcus-Thermus", "Chloroflexi", "Basidiomycota"), name = NULL) +
  scale_fill_manual(values = c("#004949", "#693A91"), labels = c("Bacteria", "Fungi")) +
  scale_x_continuous(name = bquote("Log Fold Change per"~italic(S.)~italic(musiva)*" Copy")) +
  geom_text(data = tex, aes(label = label, x = x, y = y), inherit.aes = F, size = 3) +
  geom_hline(yintercept = 7.5, linetype = "dashed") +
  theme(legend.position = "none", axis.title = element_text(size = 9))-> fig.5a.new

tex <- data.frame(label = c("Rhizosphere", "Leaf Surface"),
                  x = rep(0.0005,2),
                  y = c(1, 2))
plot_data %>% 
  filter(sample_type == "LW" | sample_type == "RH") %>%
  dplyr::mutate(int = fct_reorder(int, b)) %>%
  dplyr::mutate(int = fct_reorder(int, sample_type, .desc = T)) %>%
  ggplot(aes(x = b, y = int, fill = Kingdom)) +
  geom_bar(stat = "identity") +
  geom_errorbarh(aes(xmin = b - se$Copy_number_per_ul, xmax = b + se$Copy_number_per_ul), height = 0.2) +
  geom_vline(xintercept = 0)  +
  scale_y_discrete(labels = c("Aphelidiomycota", "Deinococcus-Thermus", "Proteobacteria", 
                              "Actinobacteria", "BRC1", "Bacteroidetes", "Latescibacteria", "Nitrospirae", "Thaumarchaeota", 
                              "Chytridiomycota"),
                   name = NULL) +
  scale_fill_manual(values = c("#920001", "#004949", "#693A91"), labels = c("Archaea", "Bacteria", "Fungi")) +
  scale_x_continuous(name = bquote("Log Fold Change per"~italic(S.)~italic(musiva)*" Copy"), breaks = c(-0.001, -0.0005, 0, 0.0005)) +
  geom_text(data = tex, aes(label = label, x = x, y = y), inherit.aes = F, size = 3) +
  geom_hline(yintercept = 1.5, linetype = "dashed") +
  theme(legend.position = "none", axis.title = element_text(size = 9))-> fig.5b.new

plot_data2 <- rbind(sig_tab_16S_SA, sig_tab_ITS_SA) %>% filter(level == 4)
plot_data2$Genus <- sapply(strsplit(as.character(plot_data2$Genus), "__"), `[`, 2)

plot_data2 %>% 
  filter(Genus != "unidentified") %>%
  dplyr::mutate(Genus = fct_reorder(Genus, b)) %>%
  ggplot(aes(x = b, y = Genus, fill = Kingdom)) +
  geom_bar(stat = "identity") +
  geom_errorbarh(aes(xmin = b - se$stearic.acid, xmax = b + se$stearic.acid), height = 0.2) +
  geom_vline(xintercept = 0)  +
  scale_y_discrete(name = NULL) +
  scale_fill_manual(values = c("#004949", "#693A91"), labels = c("Bacteria", "Fungi")) +
  scale_x_continuous(name = bquote("Log Fold Change per stearic acid"~mu*"g"~g^-1)) +
  theme(legend.position = "none", axis.text.y = element_text(face = "italic"))-> fig.5c.new

plots <- align_plots(fig.5c.new, fig.5a.new, align = 'v', axis = 'l')
top_row <- plot_grid(plots[[2]], fig.5b.new, labels = c('A', 'B'))
prow <- plot_grid(top_row, plots[[1]], labels = c('', 'C'), ncol = 1, rel_heights = c(10,12))
leg <- get_legend(fig.5b.new + theme(legend.position = "bottom", legend.justification = "center"))
prow3 <- plot_grid(prow, leg, ncol = 1, rel_heights = c(1, 0.08))

ggsave("~/Downloads/Figure5.pdf", width = 7.5, height = 6.5, units = "in", dpi = 600)
prow3
dev.off()

fig.5a +
  theme(axis.text = element_text(size = 20), axis.title = element_text(size = 24), text = element_text(size = 24)) -> p

fig.5b+
  theme(legend.position = "none", axis.text = element_text(size = 20), axis.title = element_text(size = 24)) -> p2

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figba_poster.pdf", width = 9, height = 7, units = "in", dpi = 600)
p
dev.off()


myplots <- list()
for(i in levels(as.factor(data_16S_tss@sam_data$sample_type))){
  sub <- subset_samples(data_16S_tss, sample_type %in% i)
  set.seed(01221990)
  ordu <- ordinate(sub, "PCoA", "bray", weighted = T)
  p <- plot_ordination(sub, ordu)
  p2 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21,24),
                       labels = c(expression(italic("noninfected")), 
                                  expression(italic("infected")))) +
    ggtitle(i) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21, size = 4))) 
  myplots[[i]] <- p2 + theme(legend.position = "none")
  print(i)
}

leg <- get_legend(p2 + theme(legend.position = "bottom", legend.box = "vertical", legend.justification = "center"))
grid_16S <- plot_grid(plotlist = myplots, align = "vh")
beta_16S2 <- plot_grid(grid_16S, leg, ncol = 1, rel_heights = c(1, 0.1))

tiff("~/Desktop/BlountCountyAssembly/Output/Figures/Presentation/Beta_16S.tiff", res = 100, height = 8, width = 8, units = "in", bg = "#DDDDDD")
beta_16S2
dev.off()

# ITS -------------------------------------------------------
myplots_ITS <- list()
for(i in levels(as.factor(data_ITS@sam_data$Description))){
  sub <- subset_samples(data_ITS, Description %in% i)
  ordu <- ordinate(sub, "PCoA", "bray", weighted = T)
  p <- plot_ordination(sub, ordu)
  p2 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = HostSpecies, fill = SampleTime), size = 2, alpha = 0.5) +
    scale_shape_manual(values = c(21,24),
                       labels = c(expression(italic("P. deltoides")), 
                                  expression(italic("P. trichocarpa")))) +
    ggtitle(i) +
    theme(legend.text = element_text(hjust = 0)) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) 
  myplots_ITS[[i]] <- p2 + theme(legend.position = "none")
  print(i)
  print(ordu$stress)
}

leg <- get_legend(p2 + theme(legend.position = "right"))
grid_ITS <- plot_grid(plotlist = myplots_ITS, align = "vh")
beta_ITS <- plot_grid(grid_ITS, leg, ncol = 2, rel_widths = c(1, 0.3))

tiff("~/Desktop/BlountCountyAssembly/Output/Figures/Presentation/Beta_ITS.tiff", res = 600, height = 6, width = 12, units = "in")
beta_ITS
dev.off()

# Figure 2 ---------------------------------------------
pgrid <- plot_grid(grid_16S, grid_ITS, labels = "AUTO", align = "vh", ncol = 1)
beta_all <- plot_grid(pgrid, leg, ncol = 2, rel_widths = c(1, 0.3))









ordu <- ordinate(data_16S_tss %>% subset_samples(sample_type == "RE"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_16S_tss_RE %>% subset_samples(sample_type == "RE"), ordu)
p1 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

ordu <- ordinate(data_16S_tss  %>% subset_samples(sample_type == "RH"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_16S_tss  %>% subset_samples(sample_type == "RH"), ordu)
p2 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

ordu <- ordinate(data_ITS_tss  %>% subset_samples(sample_type == "RE"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_ITS_tss  %>% subset_samples(sample_type == "RE"), ordu)
p3 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

ordu <- ordinate(data_ITS_tss  %>% subset_samples(sample_type == "RH"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_ITS_tss  %>% subset_samples(sample_type == "RH"), ordu)
p4 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

prow <- plot_grid(p1, p2, p3, p4,
                  align = "vh", ncol = 2, labels = "AUTO")
leg <- get_legend(p4 + theme(legend.position = "bottom", legend.justification = "center", legend.box = "vertical"))
prow2 <- plot_grid(prow, leg, ncol = 1, rel_heights = c(1, 0.1))

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure4.pdf", width = 7.5, height = 7, units = "in", dpi = 600)
prow2
dev.off()

```

# microbiome - metabolome interaction
```{r}
for(i in c("LE", "RE")){
  sub <- subset_samples(data_16S_tss, sample_type %in% i)
  dist <- distance(sub, "bray")
  dist_m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -local_sep, -Copy_number_per_ul, -Infection_2020) %>%
    dist("euclidean")
  mod <- vegan::mantel(dist, dist_m, method = "spearman", permutations = 9999, na.rm = TRUE)
  print(mod)
} # Leaf endosphere is significantly related to metabolome

for(i in c("P. deltoides", "P. trichocarpa")){
  sub <- subset_samples(data_16S_tss, sample_type == "LE" & species %in% i)
  dist <- distance(sub, "bray")
  dist_m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -local_sep, -Copy_number_per_ul, -Infection_2020) %>%
    dist("euclidean")
  mod <- vegan::mantel(dist, dist_m, method = "spearman", permutations = 9999, na.rm = TRUE)
  print(mod)
} # nothing sig.

for(i in c("LE", "RE")){
  sub <- subset_samples(data_ITS_tss, sample_type %in% i)
  dist <- distance(sub, "bray")
  dist_m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020) %>%
    dist("euclidean")
  mod <- vegan::mantel(dist, dist_m, method = "spearman", permutations = 9999, na.rm = TRUE)
  print(mod)
} # Leaf endosphere is significantly related to metabolome

for(i in c("P. deltoides", "P. trichocarpa")){
  sub <- subset_samples(data_ITS_tss, sample_type == "LE" & species %in% i)
  dist <- distance(sub, "bray")
  dist_m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020) %>%
    dist("euclidean")
  mod <- vegan::mantel(dist, dist_m, method = "spearman", permutations = 9999, na.rm = TRUE)
  print(mod)
} # nothing sig.


# overall LE 16S
sub <- subset_samples(data_16S_tss, sample_type == "LE")
dist <- distance(sub, "bray")
m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020, -local_sep) 

set.seed(01221990)
x <- vegan::dbrda(dist ~ HCH.deltoidin + X14.70.284.269.glucoside + salicyl.alcohol + benzyl.glucoside + X1.2.4.benzenetriol + o.cresol + X14.85.168.glycoside + 
                    HCH.salicortin + caffeic.acid + maleic.acid, data = m, scale = T)
set.seed(01221990)
anova(x) # 27%
# trich LE 16S
sub <- subset_samples(data_16S_tss, sample_type == "LE" & species == "P. trichocarpa")
dist <- distance(sub, "bray")
m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020, -local_sep) 

set.seed(01221990)
x <- vegan::dbrda(dist ~ X2.5.dihydroxybenzoic.acid + X13.98.445.204.105 + X14.10.445 + myo.inositol + X3.hydroxybenzoic.acid + sucrose + azelaic.acid + 
                    pentadecanoic.acid + salicylaldehyde + benzoic.acid, data = m, scale = T)
set.seed(01221990)
anova(x)


# Overall LE ITS
sub <- subset_samples(data_ITS_tss, sample_type == "LE")
dist <- distance(sub, "bray")
m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020) 

set.seed(01221990)
x <- vegan::dbrda(dist ~ glyceric.acid + erythronic.acid + HCH.deltoidin + X14.70.284.269.glucoside + benzyl.glucoside + HCH.salicortin + salicyl.alcohol + 
                    threonic.acid + X2.hydroxyglutaric.acid + salirepin, data = m, scale = T)
set.seed(01221990)
anova(x) #26%

# trich LE ITS
sub <- subset_samples(data_ITS_tss, sample_type == "LE" & species == "P. trichocarpa")
dist <- distance(sub, "bray")
m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020) 

set.seed(01221990)
x <- vegan::dbrda(dist ~ galactose + ferulic.acid + a.linolenic.acid + myo.inositol + glucose + sucrose + quinic.acid + 
                    palmitic.acid + resorcinol + oleic.acid, data = m, scale = T)
set.seed(01221990)
anova(x)



sub <- subset_samples(data_ITS_tss, sample_type == "LE" & species == "P. trichocarpa")
dist <- distance(sub, "bray")
m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020) 

set.seed(01221990)
x <- vegan::dbrda(dist ~ stearic.acid + phosphate + ethyl.phosphate + salireposide + salirepin, data = m, scale = T)
set.seed(01221990)
anova(x) # 16%

sub <- subset_samples(data_16S_tss, sample_type == "LE" & species == "P. trichocarpa")
dist <- distance(sub, "bray")
m <- sub@sam_data %>% 
    data.frame %>% 
    dplyr::select(-SampleID, -sample_type, -sample.id, -sample, -genotype, -resistance, -Copy_number_per_ul, -Infection_2020) 

set.seed(01221990)
x <- vegan::dbrda(dist ~ stearic.acid + phosphate + ethyl.phosphate + salireposide + salirepin, data = m, scale = T)
set.seed(01221990)
anova(x) # 16%

```

# Differential abundance metabolites
```{r}
# Differentially abundant with stearic.acid -------------
df_list <- list(Phylum_16S, Class_16S, Family_16S, Genus_16S, data_16S_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "stearic.acid", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_16S_SA <- tab %>% filter(q < 0.05) # Xanthamonas increases while Acinetobacter and Cutibacterium decrease

df_list <- list(Phylum_ITS, Class_ITS, Family_ITS, Genus_ITS, data_ITS_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "stearic.acid", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_ITS_SA <- tab %>% filter(q < 0.05) # Xanthamonas increases while Acinetobacter and Cutibacterium decrease

# Differentially abundant with salirepin -------------

df_list <- list(Phylum_16S, Class_16S, Family_16S, Genus_16S, data_16S_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "salirepin", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_16S_Salirepin <- tab %>% filter(q < 0.05) # increase in Neisseriaceae and Clostridiaceae

df_list <- list(Phylum_ITS, Class_ITS, Family_ITS, Genus_ITS, data_ITS_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "salirepin", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_ITS_Salirepin <- tab %>% filter(q < 0.05) # Increase in Cyberlindnera

# Differentially abundant with ethyl phosphate -------------

df_list <- list(Phylum_16S, Class_16S, Family_16S, Genus_16S, data_16S_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "ethyl.phosphate", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_16S_ethyl.phosphate <- tab %>% filter(q < 0.05) # increase in Sphingomonas

df_list <- list(Phylum_ITS, Class_ITS, Family_ITS, Genus_ITS, data_ITS_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "ethyl.phosphate", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_ITS_ethyl.phosphate <- tab %>% filter(q < 0.05) # decrease in Basidio

# Differentially abundant with phosphate -------------

df_list <- list(Phylum_16S, Class_16S, Family_16S, Genus_16S, data_16S_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "phosphate", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_16S_phosphate <- tab %>% filter(q < 0.05) # increase in Sphingomonas

df_list <- list(Phylum_ITS, Class_ITS, Family_ITS, Genus_ITS, data_ITS_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "phosphate", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_ITS_phosphate <- tab %>% filter(q < 0.05) # decrease in Basidio

# Differentially abundant with salireposide -------------

df_list <- list(Phylum_16S, Class_16S, Family_16S, Genus_16S, data_16S_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "salireposide", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_16S_salireposide <- tab %>% filter(q < 0.05) # increase in Sphingomonas

df_list <- list(Phylum_ITS, Class_ITS, Family_ITS, Genus_ITS, data_ITS_combined)
tab <- data.frame(matrix(ncol = 5, nrow = 0))
for(i in 1:5){
  sub <- subset_samples(df_list[[i]], sample_type == "LE" & species == "P. trichocarpa")
  sub <-  prune_taxa(taxa_sums(sub) > 0, sub)
  out_16S_phylum <- ancombc(phyloseq = sub, formula = "salireposide", 
                            p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                            max_iter = 100, alpha = 0.05)
  res <- out_16S_phylum$res
  tab2 <- merge(res$q_val, res$beta, by = 0) 
  tab2$se <- res$se
  colnames(tab2) <- c ("OTU", "q", "b", "se")
  tab2$level <- i
  rownames(tab2) <- paste0(rownames(tab2), i)
  tab2 <- merge(tab2, df_list[[i]]@tax_table, by.x = "OTU", by.y = 0, all.x = T)
  tab <- bind_rows(tab, tab2)
}
sig_tab_ITS_salireposide <- tab %>% filter(q < 0.05) # decrease in Basidio

```



```{r}
# PCA -----------------------------------------



metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/RootMetabolites.xlsx")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "RE")

metabolites <- merge(metabolites, 
                     metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T)
metabolites <- merge(metabolites, sep %>% dplyr::select(Copy_number_per_ul, SampleID), by = "SampleID", all.x = T)
row.names(metabolites) <- metabolites$SampleID
metabolites <- filter(metabolites, SampleID != "S71" & SampleID != "S60" & SampleID != "S62")
metabolites <- metabolites %>% dplyr::select(-SampleID)

#1-149 are metabolites
metabolites$species <- "trich"
metabolites$species[metabolites$resistance == "deltoides"] <- "delt"
set.seed(01221990)
ad <- vegan::adonis(metabolites[1:149] ~ metabolites$species)
ad

trichs <- filter(metabolites, species == "trich")
set.seed(01221990)
ad <- vegan::adonis(trichs[1:149] ~ trichs$resistance)
ad

set.seed(01221990)
ad <- vegan::adonis(trichs[1:149] ~ trichs$Genotype)
ad

set.seed(01221990)
ad <- vegan::adonis(trichs[1:149] ~ trichs$Infection_2020)
ad

set.seed(01221990)
ad <- vegan::adonis(trichs[1:149] ~ trichs$Copy_number_per_ul)
ad




# bar chart of metabolites by infection trichocarpa ----------------

metabolites_long <- gather(data = metabolites, key = "metabolite", value = "value", -Genotype, -resistance, -Infection_2020)
metabolites_long$value[metabolites_long$value < 0.2] <- 0.1

metabolites_long_clean <- metabolites_long %>% filter(metabolite != "14.70 284 269 glucoside",
                                                      metabolite != "17.16 363 273 257",
                                                      metabolite != "9.82 269 284",
                                                      metabolite != "HCH-deltoidin")
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long_clean$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(Infection_2020 ~ value, data = sub %>% filter(resistance != "deltoides"), family = "binomial")
  p <- summary(a)$coefficients[2,4]
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long_clean$metabolite))))
  metabolite = i
  print(i)
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
} # No significance, 18 sig. if don't adjust p

# relating metabolites with qPCR -----------------------------

metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/RootMetabolites.xlsx")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "RE")

metabolites <- merge(metabolites, 
                     metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T)
metabolites <- merge(metabolites, sep %>% dplyr::select(Copy_number_per_ul, SampleID), by = "SampleID", all.x = T)
row.names(metabolites) <- metabolites$SampleID
metabolites <- filter(metabolites, SampleID != "S71" & SampleID != "S60" & SampleID != "S62")
metabolites <- metabolites %>% dplyr::select(-SampleID)

metabolites_long <- gather(data = metabolites, key = "metabolite", value = "value", -Genotype, -resistance, -Infection_2020, -Copy_number_per_ul)
metabolites_long$value[metabolites_long$value < 0.2] <- 0.1

tab <- data.frame(matrix(ncol = 4))
colnames(tab) <- c("metabolite", "rho", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i & resistance != "deltoides")
  x <- cor.test(sub$value, sub$Copy_number_per_ul, method = "spearman")
  rho <- x$estimate
  p <- x$p.value
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(rho, 4), round(p, 4), round(p.adj, 4)))
}# none sig., 14 sig. if non adjusted

tab <- data.frame(matrix(ncol = 4))
colnames(tab) <- c("metabolite", "rho", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  x <- cor.test(sub$value, sub$Copy_number_per_ul, method = "spearman")
  rho <- x$estimate
  p <- x$p.value
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(rho, 4), round(p, 4), round(p.adj, 4)))
} #none sig. at p < 0.05, 1 sig at p < 0.1, and 2 sig. if we don't p adjust

tab <- data.frame(matrix(ncol = 4))
colnames(tab) <- c("metabolite", "rho", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  x <- cor.test(sub$value, log10(sub$Copy_number_per_ul), method = "pearson")
  rho <- x$estimate
  p <- x$p.value
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(rho, 4), round(p, 4), round(p.adj, 4)))
} #none sig. at p < 0.05, 1 sig at p < 0.1, and 2 sig. if we don't p adjust

# Just salicilates ------------------------
points <- metabolites_long %>%
  filter(metabolite %in% c("salicin", "salicyl alcohol", "salicylic acid")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148")))

metabolites_long %>%
  filter(metabolite %in% c("salicin", "salicyl alcohol", "salicylic acid")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(Genotype, resistance, metabolite) %>%
  dplyr::summarise(mean = mean(value), se = sd(value)/sqrt(n())) %>%
  ggplot(aes(x = Genotype, y = mean, fill = resistance)) +
    geom_bar(stat = "identity") +
    geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2, color = "black") +
    geom_jitter(data = points, aes(x = Genotype, y = value, fill = resistance, shape = as.factor(as.character(Infection_2020))), 
                alpha = 0.5, inherit.aes = F) +
    scale_x_discrete(name = NULL) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_y_continuous(name = bquote(~mu*g~g^-1*" DW (sorbitol equivlents)")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    facet_wrap(~metabolite, scale = "free") +
    theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1), legend.position = "none") -> fig.2b

points <- metabolites_long %>%
  filter(metabolite %in% c("salicin", "HCH-salicortin", "salicylic acid", "salicortin", "iso-salicin", "salicylaldehyde", "salicyl alcohol")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148")))

metabolites_long %>%
  filter(metabolite %in% c("salicin", "HCH-salicortin", "salicylic acid", "salicortin", "iso-salicin", "salicylaldehyde", "salicyl alcohol")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(Genotype, resistance, metabolite) %>%
  dplyr::summarise(mean = mean(value), se = sd(value)/sqrt(n())) %>%
  ggplot(aes(x = Genotype, y = mean, fill = resistance)) +
    geom_bar(stat = "identity") +
    geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2, color = "black") +
    geom_jitter(data = points, aes(x = Genotype, y = value, fill = resistance, shape = as.factor(as.character(Infection_2020))), 
                alpha = 0.5, inherit.aes = F) +
    scale_x_discrete(name = NULL) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_y_continuous(name = bquote(~mu*g~g^-1*" DW (sorbitol equivlents)")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    facet_wrap(~metabolite, scale = "free") +
    theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1), legend.position = "none") -> p


fig2 <- plot_grid(fig.2a, fig.2b, ncol = 1, labels = "AUTO", align = "vh", axis = "l")

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure2.pdf", width = 7.5, height = 6.5, units = "in", dpi = 600)
fig2
dev.off()

metabolites_long$species <- "trich"
metabolites_long$species[metabolites_long$resistance == "deltoides"] <- "delt"

a <- aov(log(value +1) ~ species, metabolites_long %>% filter(metabolite == "salicyl alcohol"))
summary(a)
a <- aov(log(value +1) ~ species, metabolites_long %>% filter(metabolite == "salicin"))
summary(a)
a <- aov(log(value +1) ~ species, metabolites_long %>% filter(metabolite == "salicylic acid"))
summary(a)

a <- lm(log(value +1) ~ Copy_number_per_ul, metabolites_long %>% filter(metabolite == "salicyl alcohol" & species == "trich"))
summary(a)
a <- lm(log(value +1) ~ Copy_number_per_ul, metabolites_long %>% filter(metabolite == "salicin" & species == "trichocarpa"))
summary(a)
a <- lm(log(value +1) ~ Copy_number_per_ul, metabolites_long %>% filter(metabolite == "salicylic acid" & species == "trichocarpa"))
summary(a)

a <- aov(log(value +1) ~ as.factor(as.character(Infection_2020))*resistance, metabolites_long %>% filter(metabolite == "salicyl alcohol" & species == "trich"))
summary(a)
a <- aov(log(value +1) ~ Infection_2020, metabolites_long %>% filter(metabolite == "salicin" & species == "trichocarpa"))
summary(a)
a <- aov(log(value +1) ~ Infection_2020, metabolites_long %>% filter(metabolite == "salicylic acid" & species == "trichocarpa"))
summary(a)

metabolites_long %>% 
  filter(metabolite %in% c("salicyl alcohol", "salicin", "salicylic acid")) %>%
  group_by(metabolite, species) %>%
  dplyr::summarise(mean = mean(value))

metabolites_long %>% 
  filter(metabolite == "HCH-salicortin" & species == "trichocarpa") %>%
  group_by(metabolite, Infection_2020) %>%
  dplyr::summarise(mean = mean(value))

# Changing by species --------------------------------
metabolites_long$species <- "trichocarpa"
metabolites_long$species[metabolites_long$resistance == "deltoides"] <- "deltoides"

tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(log10(value +1) ~ species, data = sub, family = Gamma())
  p <- summary(a)$coefficients[2,4]
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  print(i)
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
sig_tab <- tab %>% filter(p.adj < 0.05)

p_data <- metabolites_long %>%
  filter(metabolite %in% sig_tab$metabolite) %>%
  group_by(metabolite, species) %>%
  dplyr::summarise(mean = mean(value))

p_data %>% 
  spread(key = species, value = mean) %>%
  mutate(y = log2(deltoides/trichocarpa)) %>%
  dplyr::mutate(metabolite = fct_reorder(metabolite, as.numeric(y))) %>%
  ggplot(aes(x = reorder(metabolite, y), y = y)) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = 0, ymax = Inf), fill = "#76b5d6", alpha = 0.5) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 0), fill = "#B6D084", alpha = 0.5) +
  geom_bar(stat = "identity")+
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log2 fold change\n(deltoides/trichocarpa)")) +
  theme(axis.text.y = element_text(size = 8)) +
  coord_flip() -> fig.S1b

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_S1.pdf", width = 6.5, height = 7.5, units = "in", dpi = 600)
fig.S1
dev.off()


metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/RootMetabolites.xlsx")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "RE")

metabolites <- merge(metabolites, metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T)
row.names(metabolites) <- metabolites$SampleID
metabolites <- filter(metabolites, SampleID != c("S71", "S60", "S62"))
metabolites <- metabolites %>% dplyr::select(-SampleID)

metabolites_long <- gather(data = metabolites, key = "metabolite", value = "value", -Genotype, -resistance, -Infection_2020)
metabolites_long$value[metabolites_long$value < 0.2] <- 0.1
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(value ~resistance, data = sub %>% filter(resistance != "deltoides"), family = Gamma())
  res <- Anova(a)
  p <- res[1, 3]
  p.adj <- p.adjust(p = res[1, 3], method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
sig_tab <- tab %>% filter(p.adj < 0.05)


tab2 <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(value ~resistance, data = sub %>% filter(resistance != "deltoides"), family = Gamma())
  res <- Anova(a)
  p <- res[1, 3]
  p.adj <- p.adjust(p = res[1, 3], method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab2 <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
# no differences between resistant/suceptible

sig_tab <- tab %>% filter(p < 0.05)
p_data <- metabolites_long %>%
  filter(metabolite %in% sig_tab$metabolite & resistance != "deltoides") %>%
  dplyr::mutate(Infection_2020 = dplyr::recode(Infection_2020, `0` = "noninfected", `1` = "infected")) %>%
  group_by(metabolite, Infection_2020) %>%
  dplyr::summarise(mean = mean(value))

p_data %>% 
  spread(key = Infection_2020, value = mean) %>%
  as.data.frame() %>%
  dplyr::mutate(y = log2(noninfected/infected)) %>%
  dplyr::mutate(metabolite = fct_reorder(metabolite, y)) %>%
  ggplot(aes(x = metabolite, y = y)) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = 0, ymax = Inf), fill = "gray", alpha = 0.5) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 0), fill = "#a34d4d", alpha = 0.5) +
  geom_bar(stat = "identity")+
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log2 fold change\n(noninfected/infected)")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12), axis.text.y = element_text(size = 18), axis.title = element_text(size = 24), 
        plot.margin = unit(c(0.5, 0.5, 0.5, 2), "cm")) -> plot_metabolites_infection

tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(value ~Infection_2020, data = sub %>% filter(resistance != "deltoides"), family = Gamma())
  res <- Anova(a)
  p <- res[1, 3]
  p.adj <- p.adjust(p = res[1, 3], method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
```

# Metabolites - Leaf
```{r, warning=F, message=F, cache=F, results= F}
# PCA -----------------------------------------
metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/LeafMetabolites.xlsx")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "LE")

metabolites <- merge(metabolites, 
                     metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T)
metabolites <- merge(metabolites, sep %>% dplyr::select(Copy_number_per_ul, SampleID), by = "SampleID", all.x = T)
row.names(metabolites) <- metabolites$SampleID
metabolites <- filter(metabolites, SampleID != "S62" & SampleID != "S60" & SampleID != "S10")
metabolites <- metabolites %>% dplyr::select(-SampleID)

#1-135 are metabolites
metabolites$species <- "trich"
metabolites$species[metabolites$resistance == "deltoides"] <- "delt"
set.seed(01221990)
ad <- vegan::adonis(metabolites[1:135] ~ metabolites$species)
ad # sig. species effect

trichs <- filter(metabolites, species == "trich")
set.seed(01221990)
ad <- vegan::adonis(trichs[1:135] ~ trichs$resistance)
ad # no effect

set.seed(01221990)
ad <- vegan::adonis(trichs[1:135] ~ trichs$Genotype)
ad # genotype effect

set.seed(01221990)
ad <- vegan::adonis(trichs[1:135] ~ trichs$Infection_2020)
ad # sig effect p = 0.024, R2 = 0.034

set.seed(01221990)
trichs_noNA <- trichs %>% filter(Copy_number_per_ul > 0)
ad <- vegan::adonis(trichs_noNA[1:135] ~ trichs_noNA$Copy_number_per_ul)
ad # no sig.


ord <- prcomp(metabolites[1:135])
ord <- merge(ord$x %>% as.data.frame() %>% dplyr::select(PC1, PC2), metabolites %>% dplyr::select(Genotype, resistance, Infection_2020), by = 0)

ord %>%
  ggplot(aes(x = PC1, y = PC2, fill = resistance, shape = as.character(Infection_2020))) +
  geom_point(size = 2, color = "black") +
  labs(x = "PC1 (43%)", y = "PC2 (23%)") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), labels = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  stat_ellipse(aes(color = resistance, group = resistance)) +
  scale_color_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), guide = "none") +
  geom_text(data = data.frame(resistance = c("deltoides", "trichocarpa - resistant", "trichocarpa - susceptible"),
                              x = c(20000, 10000, 20000),
                              y = c(7000, -8000, -3000)),
            aes(x = x, y = y, label = resistance, color = resistance), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible"), inherit.aes = F) +
  scale_x_continuous(breaks = c(-10000, 0, 10000, 20000)) +
  theme(legend.title = element_blank(), legend.position = "right") +
  guides(fill = guide_legend(override.aes = list(pch = 21, size = 4))) -> fig.2a

# bar chart of metabolites by infection trichocarpa ----------------

metabolites_long <- gather(data = metabolites, key = "metabolite", value = "value", -Genotype, -resistance, -Infection_2020)
metabolites_long$value[metabolites_long$value < 0.2] <- 0.1

metabolites_long_clean <- metabolites_long %>% filter(metabolite != "14.70 284 269 glucoside",
                                                      metabolite != "species",
                                                      #metabolite != "9.82 269 284",
                                                      metabolite != "HCH-deltoidin")
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long_clean$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(Infection_2020 ~ value, data = sub %>% filter(resistance != "deltoides"), family = "binomial")
  p <- summary(a)$coefficients[2,4]
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long_clean$metabolite))))
  metabolite = i
  print(i)
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
} # No sig

# relating metabolites with qPCR -----------------------------

metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/LeafMetabolites.xlsx")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "LE")

metabolites <- merge(metabolites, 
                     metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T)
metabolites <- merge(metabolites, sep %>% dplyr::select(Copy_number_per_ul, SampleID), by = "SampleID", all.x = T)
row.names(metabolites) <- metabolites$SampleID
metabolites <- filter(metabolites, SampleID != "S10" & SampleID != "S60" & SampleID != "S62")
metabolites <- metabolites %>% dplyr::select(-SampleID)

metabolites_long <- gather(data = metabolites, key = "metabolite", value = "value", -Genotype, -resistance, -Infection_2020, -Copy_number_per_ul)
metabolites_long$value[metabolites_long$value < 0.2] <- 0.1

tab <- data.frame(matrix(ncol = 4))
colnames(tab) <- c("metabolite", "rho", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i & resistance != "deltoides")
  x <- cor.test(sub$value, sub$Copy_number_per_ul, method = "spearman")
  rho <- x$estimate
  p <- x$p.value
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(rho, 4), round(p, 4), round(p.adj, 4)))
}# none sig., 2 sig if not adjusted: erythronic acid gamma-lactone, 6-hydroxy-2-cyclohexenone alcohol, both positive

tab <- data.frame(matrix(ncol = 4))
colnames(tab) <- c("metabolite", "rho", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  x <- cor.test(sub$value, sub$Copy_number_per_ul, method = "spearman")
  rho <- x$estimate
  p <- x$p.value
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(rho, 4), round(p, 4), round(p.adj, 4)))
} #none sig

tab <- data.frame(matrix(ncol = 4))
colnames(tab) <- c("metabolite", "rho", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  x <- cor.test(sub$value, log10(sub$Copy_number_per_ul), method = "pearson")
  rho <- x$estimate
  p <- x$p.value
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(rho, 4), round(p, 4), round(p.adj, 4)))
} #none sig. at p < 0.05, 1 sig at p < 0.1, and 2 sig. if we don't p adjust

# Just salicilates ------------------------
points <- metabolites_long %>%
  filter(metabolite %in% c("salicin", "salicyl alcohol", "salicylic acid")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148")))

metabolites_long %>%
  filter(metabolite %in% c("salicin", "salicyl alcohol", "salicylic acid")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(Genotype, resistance, metabolite) %>%
  dplyr::summarise(mean = mean(value), se = sd(value)/sqrt(n())) %>%
  ggplot(aes(x = Genotype, y = mean, fill = resistance)) +
    geom_bar(stat = "identity") +
    geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2, color = "black") +
    geom_jitter(data = points, aes(x = Genotype, y = value, fill = resistance, shape = as.factor(as.character(Infection_2020))), 
                alpha = 0.5, inherit.aes = F) +
    scale_x_discrete(name = NULL) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_y_continuous(name = bquote(~mu*g~g^-1*" DW (sorbitol equivlents)")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    facet_wrap(~metabolite, scale = "free") +
    theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1), legend.position = "none") -> fig.2b

points <- metabolites_long %>%
  filter(metabolite %in% c("salicin", "HCH-salicortin", "salicylic acid", "salicortin", "iso-salicin", "salicylaldehyde", "salicyl alcohol")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148")))

metabolites_long %>%
  filter(metabolite %in% c("salicin", "HCH-salicortin", "salicylic acid", "salicortin", "iso-salicin", "salicylaldehyde", "salicyl alcohol")) %>%
  dplyr::mutate(Genotype = ordered(Genotype, c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(Genotype, resistance, metabolite) %>%
  dplyr::summarise(mean = mean(value), se = sd(value)/sqrt(n())) %>%
  ggplot(aes(x = Genotype, y = mean, fill = resistance)) +
    geom_bar(stat = "identity") +
    geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2, color = "black") +
    geom_jitter(data = points, aes(x = Genotype, y = value, fill = resistance, shape = as.factor(as.character(Infection_2020))), 
                alpha = 0.5, inherit.aes = F) +
    scale_x_discrete(name = NULL) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_y_continuous(name = bquote(~mu*g~g^-1*" DW (sorbitol equivlents)")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    facet_wrap(~metabolite, scale = "free") +
    theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1), legend.position = "none") -> p


fig2 <- plot_grid(fig.2a, fig.2b, ncol = 1, labels = "AUTO", align = "vh", axis = "l")

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure2.pdf", width = 7.5, height = 6.5, units = "in", dpi = 600)
fig2
dev.off()

metabolites_long$species <- "trich"
metabolites_long$species[metabolites_long$resistance == "deltoides"] <- "delt"

a <- aov(log(value +1) ~ species, metabolites_long %>% filter(metabolite == "salicyl alcohol"))
summary(a)
a <- aov(log(value +1) ~ species, metabolites_long %>% filter(metabolite == "salicin"))
summary(a)
a <- aov(log(value +1) ~ species, metabolites_long %>% filter(metabolite == "salicylic acid"))
summary(a)

a <- lm(log(value +1) ~ Copy_number_per_ul, metabolites_long %>% filter(metabolite == "salicyl alcohol" & species == "trich"))
summary(a)
a <- lm(log(value +1) ~ Copy_number_per_ul, metabolites_long %>% filter(metabolite == "salicin" & species == "trichocarpa"))
summary(a)
a <- lm(log(value +1) ~ Copy_number_per_ul, metabolites_long %>% filter(metabolite == "salicylic acid" & species == "trichocarpa"))
summary(a)

a <- aov(log(value +1) ~ as.factor(as.character(Infection_2020))*resistance, metabolites_long %>% filter(metabolite == "salicyl alcohol" & species == "trich"))
summary(a)
a <- aov(log(value +1) ~ Infection_2020, metabolites_long %>% filter(metabolite == "salicin" & species == "trichocarpa"))
summary(a)
a <- aov(log(value +1) ~ Infection_2020, metabolites_long %>% filter(metabolite == "salicylic acid" & species == "trichocarpa"))
summary(a)

metabolites_long %>% 
  filter(metabolite %in% c("salicyl alcohol", "salicin", "salicylic acid")) %>%
  group_by(metabolite, species) %>%
  dplyr::summarise(mean = mean(value))

metabolites_long %>% 
  filter(metabolite == "HCH-salicortin" & species == "trichocarpa") %>%
  group_by(metabolite, Infection_2020) %>%
  dplyr::summarise(mean = mean(value))

# Changing by species --------------------------------
metabolites_long$species <- "trichocarpa"
metabolites_long$species[metabolites_long$resistance == "deltoides"] <- "deltoides"

tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(log10(value +1) ~ species, data = sub, family = Gamma())
  p <- summary(a)$coefficients[2,4]
  p.adj <- p.adjust(p = p, method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  print(i)
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
sig_tab <- tab %>% filter(p.adj < 0.05)

p_data <- metabolites_long %>%
  filter(metabolite %in% sig_tab$metabolite) %>%
  group_by(metabolite, species) %>%
  dplyr::summarise(mean = mean(value))

p_data %>% 
  spread(key = species, value = mean) %>%
  mutate(y = log2(deltoides/trichocarpa)) %>%
  dplyr::mutate(metabolite = fct_reorder(metabolite, as.numeric(y))) %>%
  ggplot(aes(x = reorder(metabolite, y), y = y)) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = 0, ymax = Inf), fill = "#76b5d6", alpha = 0.5) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 0), fill = "#B6D084", alpha = 0.5) +
  geom_bar(stat = "identity")+
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log2 fold change\n(deltoides/trichocarpa)")) +
  theme(axis.text.y = element_text(size = 8)) +
  coord_flip() -> fig.S1a

fig.S1 <- plot_grid(fig.S1a, fig.S1b, labels = "AUTO", align = "vh")

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure_S1.pdf", width = 6.5, height = 7.5, units = "in", dpi = 600)
fig.S1
dev.off()


metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/LeafMetabolites.xlsx")
metadata <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t") %>% filter(sample_type == "LE")

metabolites <- merge(metabolites, metadata %>% dplyr::select(Genotype, resistance, SampleID, Infection_2020) %>% distinct(), 
                     by = "SampleID", all.x = T)
row.names(metabolites) <- metabolites$SampleID
metabolites <- filter(metabolites, SampleID != c("S10", "S60", "S62"))
metabolites <- metabolites %>% dplyr::select(-SampleID)

metabolites_long <- gather(data = metabolites, key = "metabolite", value = "value", -Genotype, -resistance, -Infection_2020)
metabolites_long$value[metabolites_long$value < 0.2] <- 0.1
tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(value ~resistance, data = sub %>% filter(resistance != "deltoides"), family = Gamma())
  res <- Anova(a)
  p <- res[1, 3]
  p.adj <- p.adjust(p = res[1, 3], method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
sig_tab <- tab %>% filter(p.adj < 0.05)


tab2 <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(value ~resistance, data = sub %>% filter(resistance != "deltoides"), family = Gamma())
  res <- Anova(a)
  p <- res[1, 3]
  p.adj <- p.adjust(p = res[1, 3], method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab2 <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
# no differences between resistant/suceptible

sig_tab <- tab %>% filter(p < 0.05)
p_data <- metabolites_long %>%
  filter(metabolite %in% sig_tab$metabolite & resistance != "deltoides") %>%
  dplyr::mutate(Infection_2020 = dplyr::recode(Infection_2020, `0` = "noninfected", `1` = "infected")) %>%
  group_by(metabolite, Infection_2020) %>%
  dplyr::summarise(mean = mean(value))

p_data %>% 
  spread(key = Infection_2020, value = mean) %>%
  as.data.frame() %>%
  dplyr::mutate(y = log2(noninfected/infected)) %>%
  dplyr::mutate(metabolite = fct_reorder(metabolite, y)) %>%
  ggplot(aes(x = metabolite, y = y)) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = 0, ymax = Inf), fill = "gray", alpha = 0.5) +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 0), fill = "#a34d4d", alpha = 0.5) +
  geom_bar(stat = "identity")+
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log2 fold change\n(noninfected/infected)")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12), axis.text.y = element_text(size = 18), axis.title = element_text(size = 24), 
        plot.margin = unit(c(0.5, 0.5, 0.5, 2), "cm")) -> plot_metabolites_infection

tab <- data.frame(matrix(ncol = 3))
colnames(tab) <- c("metabolite", "p", "p.adj")
for(i in levels(as.factor(metabolites_long$metabolite))){
  sub <- subset(metabolites_long, metabolite %in% i)
  a <- glm(value ~Infection_2020, data = sub %>% filter(resistance != "deltoides"), family = Gamma())
  res <- Anova(a)
  p <- res[1, 3]
  p.adj <- p.adjust(p = res[1, 3], method = "fdr", n = length(levels(as.factor(metabolites_long$metabolite))))
  metabolite = i
  tab <- rbind(tab, c(metabolite, round(p, 4), round(p.adj, 4)))
}
```



```

# Microbiome 2 (beta diversity)
```{r}
# Download data --------------
data_16S <- qza_to_phyloseq_ITS(features = "~/Desktop/BC_Septoria_Study/Data/16S/QZAs/table-16S_merge_021121.qza",  
                              taxonomy = "~/Desktop/BC_Septoria_Study/Data/16S/QZAs/taxonomy_merged_021121.qza")

metadata_16S <- read_delim("~/Desktop/BC_Septoria_Study/Data/16S/metadata_16S.txt", delim = "\t")
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t")
metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/RootMetabolites.xlsx")
scoring <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 

metadata_16S <- merge(metadata_16S, sep %>% dplyr::select(sample_type, SampleID, Copy_number_per_ul), by = c("sample_type", "SampleID"), all.x = T)
metadata_16S <- merge(metadata_16S, metabolites, by = "SampleID")
metadata_16S <- merge(metadata_16S, scoring %>% dplyr::select(SampleID, Infection_2020), by = "SampleID")

sampledata_16S <- sample_data(metadata_16S) #import into phyloseq sample data object
sample_names(sampledata_16S) <- sampledata_16S$`sample-id` # make sure names are the same as the OTU tables
data_16S <- merge_phyloseq(data_16S, sampledata_16S)

data_16S <- subset_taxa(data_16S, Kingdom == "D_0__Bacteria" | Kingdom == "D_0__Archaea")
data_16S <- subset_taxa(data_16S, Order != "D_3__Chloroplast")
data_16S <- subset_taxa(data_16S, Family != "D_4__Mitochondria")

data_16S_RE <- prune_samples(sample_sums(data_16S)>=8000, data_16S) %>% subset_samples(sample_type == "RE")
data_16S_RE_counts <- data_16S_RE
data_16S_RH <- prune_samples(sample_sums(data_16S)>=10000, data_16S) %>% subset_samples(sample_type == "RH")
data_16S_RH_counts <- data_16S_RH

data_16S_combined <- merge_phyloseq(data_16S_RE, data_16S_RH)
data_16S_combined <- subset_samples(data_16S_combined, SampleID != "S71" & SampleID != "S60" & SampleID != "S62")

data_ITS <- qza_to_phyloseq_ITS(features = "~/Desktop/BC_Septoria_Study/Data/ITS/QZAs/table-ITS_merge_021121.qza",  
                              taxonomy = "~/Desktop/BC_Septoria_Study/Data/ITS/QZAs/taxonomy-ITS_021121.qza")

metadata_ITS <- read_delim("~/Desktop/BC_Septoria_Study/Data/ITS/metadata_ITS.txt", delim = "\t")
sep <- read_delim("~/Desktop/BC_Septoria_Study/Data/Sep_qPCR.txt", delim = "\t")
metabolites <- read_excel("~/Desktop/BC_Septoria_Study/Data/RootMetabolites.xlsx")
scoring <- read_excel("~/Desktop/BC_Septoria_Study/Data/Scoring.xlsx") 

metadata_ITS <- merge(metadata_ITS, sep %>% dplyr::select(sample_type, SampleID, Copy_number_per_ul), by = c("sample_type", "SampleID"), all.x = T)
metadata_ITS <- merge(metadata_ITS, metabolites, by = "SampleID")
metadata_ITS <- merge(metadata_ITS, scoring %>% dplyr::select(SampleID, Infection_2020), by = "SampleID")

sampledata_ITS <- sample_data(metadata_ITS) #import into phyloseq sample data object
sample_names(sampledata_ITS) <- sampledata_ITS$`sample-id` # make sure names are the same as the OTU tables
data_ITS <- merge_phyloseq(data_ITS, sampledata_ITS)

data_ITS <- subset_taxa(data_ITS, Kingdom == "k__Fungi")
data_ITS_RE <- prune_samples(sample_sums(data_ITS)>=3000, data_ITS) %>% subset_samples(sample_type == "RE")
data_ITS_RE_counts <- data_ITS_RE
data_ITS_RH <- prune_samples(sample_sums(data_ITS)>=20000, data_ITS) %>% subset_samples(sample_type == "RH")
data_ITS_RH_counts <- data_ITS_RH

data_ITS_combined <- merge_phyloseq(data_ITS_RE, data_ITS_RH)
data_ITS_combined <- subset_samples(data_ITS_combined, SampleID != "S71" & SampleID != "S60" & SampleID != "S62")

# Beta Diversity by genotype, resitance, and host species and infection --------------

data_16S_tss <- microbiome::transform(data_16S_combined, transform = "compositional")
dist <- distance(data_16S_tss, "bray")
df_16S <- data.frame(sample_data(data_16S_tss))
set.seed(01221990)
ad <- vegan::adonis(dist ~ sample_type*resistance*Infection_2020, df_16S) # large differences by sample type

data_ITS_tss <- microbiome::transform(data_ITS_combined, transform = "compositional")
dist <- distance(data_ITS_tss, "bray")
df_ITS <- data.frame(sample_data(data_ITS_tss))
set.seed(01221990)
ad <- vegan::adonis(dist ~ sample_type*genotype*Infection_2020, df_ITS) # large differences by sample type, small host species effect

# plot for sample type effect

ordu <- ordinate(data_16S_tss, "PCoA", "bray", weighted = T)
p <- plot_ordination(data_16S_tss, ordu)
p2 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = species, fill = sample_type), size = 2) +
    scale_shape_manual(values = c(21,24)) +
    scale_fill_manual(values = c("green", "brown"), labels = c("Root Endosphere", "Rhizosphere")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
  theme(legend.position = "bottom", legend.title = element_blank())

ordu <- ordinate(data_ITS_tss, "PCoA", "bray", weighted = T)
p3 <- plot_ordination(data_ITS_tss, ordu)
p4 <- ggplot(p3$data, p3$mapping) +
    geom_point(aes(shape = species, fill = sample_type), size = 2) +
    scale_shape_manual(values = c(21,24)) +
    scale_fill_manual(values = c("green", "brown"), labels = c("Root Endosphere", "Rhizosphere")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
  theme(legend.position = "bottom", legend.title = element_blank())

prow <- plot_grid(p2 + theme(legend.position = "none"),
                  p4 + theme(legend.position = "none"),
                  align = "vh", ncol = 2, labels = "AUTO")
leg <- get_legend(p4 + theme(legend.justification = "center"))
prow2 <- plot_grid(prow, leg, ncol = 1, rel_heights = c(1, 0.1))

ggsave("~/Desktop/BC_Septoria_Study/Figures/FigureS5.pdf", width = 6.5, height = 4, units = "in", dpi = 600)
prow2
dev.off()

for(i in levels(as.factor(as.character(data_16S_tss@sam_data$sample_type)))){
  sub <- subset_samples(data_16S_tss, sample_type %in% i)
  dist <- distance(sub, "bray")
  df <- data.frame(sample_data(sub))
  set.seed(01221990)
  print(i)
  print(vegan::adonis(dist ~ species, df)) 
  sub.sub <- subset_samples(sub, species == "P. trichocarpa")
  dist <- distance(sub.sub, "bray")
  df <- data.frame(sample_data(sub.sub))
  set.seed(01221990)
  print(vegan::adonis(dist ~ genotype*Infection_2020, df)) 
  print(vegan::adonis(dist ~ resistance*Infection_2020, df))
  sub.sub.sub <- subset_samples(sub, Infection_2020 == "0")
  dist <- distance(sub.sub.sub, "bray")
  df <- data.frame(sample_data(sub.sub.sub))
  set.seed(01221990)
  print(vegan::adonis(dist ~ genotype, df)) 
  print(vegan::adonis(dist ~ resistance, df))
  sub.sub.sub <- subset_samples(sub, Infection_2020 == "1")
  dist <- distance(sub.sub.sub, "bray")
  df <- data.frame(sample_data(sub.sub.sub))
  set.seed(01221990)
  print(vegan::adonis(dist ~ genotype, df)) 
  print(vegan::adonis(dist ~ resistance, df))
}

for(i in levels(as.factor(as.character(data_ITS_tss@sam_data$sample_type)))){
  sub <- subset_samples(data_ITS_tss, sample_type %in% i)
  dist <- distance(sub, "bray")
  df <- data.frame(sample_data(sub))
  set.seed(01221990)
  print(i)
  print(vegan::adonis(dist ~ species, df)) 
  sub.sub <- subset_samples(sub, species == "P. trichocarpa")
  dist <- distance(sub.sub, "bray")
  df <- data.frame(sample_data(sub.sub))
  set.seed(01221990)
  print(vegan::adonis(dist ~ genotype*Infection_2020, df)) 
  print(vegan::adonis(dist ~ resistance*Infection_2020, df))
  sub.sub.sub <- subset_samples(sub, Infection_2020 == "0")
  dist <- distance(sub.sub.sub, "bray")
  df <- data.frame(sample_data(sub.sub.sub))
  set.seed(01221990)
  print(vegan::adonis(dist ~ genotype, df)) 
  print(vegan::adonis(dist ~ resistance, df))
  sub.sub.sub <- subset_samples(sub, Infection_2020 == "1")
  dist <- distance(sub.sub.sub, "bray")
  df <- data.frame(sample_data(sub.sub.sub))
  set.seed(01221990)
  print(vegan::adonis(dist ~ genotype, df)) 
  print(vegan::adonis(dist ~ resistance, df))
}
  
# plots 
ordu <- ordinate(data_16S_tss %>% subset_samples(sample_type == "RE"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_16S_tss_RE %>% subset_samples(sample_type == "RE"), ordu)
p1 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

ordu <- ordinate(data_16S_tss  %>% subset_samples(sample_type == "RH"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_16S_tss  %>% subset_samples(sample_type == "RH"), ordu)
p2 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

ordu <- ordinate(data_ITS_tss  %>% subset_samples(sample_type == "RE"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_ITS_tss  %>% subset_samples(sample_type == "RE"), ordu)
p3 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

ordu <- ordinate(data_ITS_tss  %>% subset_samples(sample_type == "RH"), "PCoA", "bray", weighted = T)
p <- plot_ordination(data_ITS_tss  %>% subset_samples(sample_type == "RH"), ordu)
p4 <- ggplot(p$data, p$mapping) +
    geom_point(aes(shape = as.character(Infection_2020), fill = resistance), size = 2) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), label = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("PCoA 1 (", round(100*as.data.frame(ordu$values)[1,2], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("PCoA 2 (", round(100*as.data.frame(ordu$values)[2,2], 1)), "%)")) +
    guides(fill = guide_legend(override.aes = list(pch = 21))) +
    theme(legend.position = "none", legend.title = element_blank())

prow <- plot_grid(p1, p2, p3, p4,
                  align = "vh", ncol = 2, labels = "AUTO")
leg <- get_legend(p4 + theme(legend.position = "bottom", legend.justification = "center", legend.box = "vertical"))
prow2 <- plot_grid(prow, leg, ncol = 1, rel_heights = c(1, 0.1))

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure4.pdf", width = 7.5, height = 7, units = "in", dpi = 600)
prow2
dev.off()


x <- data_16S_tss %>% subset_samples(sample_type == "RE") %>% subset_taxa(Phylum == "D_1__Deinococcus-Thermus" |
                                                                            Phylum == "D_1__Rokubacteria" |
                                                                            Phylum == "D_1__Elusimicrobia" |
                                                                            Phylum == "D_1__Nitrospirae" |
                                                                            Phylum == "D_1__WS2" |
                                                                            Phylum == "D_1__Patescibacteria" |
                                                                            Phylum == "D_1__Fibrobacteres")



# Beta Diversity by sep --------------

sub <- subset_samples(data_16S_tss, Copy_number_per_ul > 0)

dist <- distance(sub, "bray")
df_16S <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ sample_type*Copy_number_per_ul, df_16S) # Interaction

dist <- distance(sub %>% subset_samples(sample_type == "LE"), "bray")
df_16S <- data.frame(sample_data(sub %>% subset_samples(sample_type == "LE")))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Copy_number_per_ul, df_16S) # nothing

dist <- distance(sub %>% subset_samples(sample_type == "LW"), "bray")
df_16S <- data.frame(sample_data(sub %>% subset_samples(sample_type == "LW")))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Copy_number_per_ul, df_16S) # nothing

dist <- distance(sub %>% subset_samples(sample_type == "RE"), "bray")
df_16S <- data.frame(sample_data(sub %>% subset_samples(sample_type == "RE")))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Copy_number_per_ul, df_16S) # almost

dist <- distance(sub %>% subset_samples(sample_type == "RH"), "bray")
df_16S <- data.frame(sample_data(sub %>% subset_samples(sample_type == "RH")))
set.seed(01221990)
ad <- vegan::adonis(dist ~ Copy_number_per_ul, df_16S) # nothing

sub <- subset_samples(data_ITS_tss, Copy_number_per_ul > 0)
dist <- distance(sub, "bray")
df_ITS <- data.frame(sample_data(sub))
set.seed(01221990)
ad <- vegan::adonis(dist ~ sample_type*Copy_number_per_ul, df_ITS) # large differences by sample type, small host species effect

# Microbiome influenced by the metabolome --------------------
# Mantel test
data_16S_tss_RE <- subset_samples(data_16S_tss, sample_type == "RE")
df_16S_RE <- data.frame(sample_data(data_16S_tss_RE))

metabolites <- df_16S_RE[,10:158]
dist <- distance(data_16S_tss_RE, "bray")
dist_m <- dist(metabolites, "euclidean")

mod <- vegan::mantel(dist, dist_m, method = "pearson", permutations = 9999, na.rm = TRUE)

data_ITS_tss_RE <- subset_samples(data_ITS_tss, sample_type == "RE")
df_ITS_RE <- data.frame(sample_data(data_ITS_tss_RE))

metabolites <- df_ITS_RE[,10:158]
dist <- distance(data_ITS_tss_RE, "bray")
dist_m <- dist(metabolites, "euclidean")

mod <- vegan::mantel(dist, dist_m, method = "spearman", permutations = 9999, na.rm = TRUE)

# dbRDA
dist <- distance(data_16S_tss_RE, "bray")

set.seed(01221990)
x <- vegan::dbrda(dist ~ salicylaldehyde + trichocarpinene + helicin + iso.salicin + salirepin + gentisyl.alcohol.5.O.glucoside + salicin + gentisic.acid.2.O.glucoside +
                    gentisic.acid.5.O.glucoside + X2.5.dihydroxybenzoic.acid.methyl.ester.5.O.glucoside + X3.methoxysalicylic.acid.B.D.glucoside +
                    tremuloidin + trichocarpin + salicyl.salicylic.acid.2.O.glucoside + salireposide + a.salicyloylsalicin + salicyloyl.coumaroyl.glucoside +
                    X2.5.dihydroxybenzoic.acid.methyl.ester.2.O.glucoside + HCH.deltoidin + tremulacin + HCH.salicortin + caffeoyltremuloidin +
                    salicyl.alcohol + salicylic.acid + salicortin, data = as.data.frame(metabolites), scale = T)
set.seed(01221990)
anova(x)

dist <- distance(data_ITS_tss_RE, "bray")

set.seed(01221990)
x <- vegan::dbrda(dist ~ salicylaldehyde + trichocarpinene + helicin + iso.salicin + salirepin + gentisyl.alcohol.5.O.glucoside + salicin + gentisic.acid.2.O.glucoside +
                    gentisic.acid.5.O.glucoside + X2.5.dihydroxybenzoic.acid.methyl.ester.5.O.glucoside + X3.methoxysalicylic.acid.B.D.glucoside +
                    tremuloidin + trichocarpin + salicyl.salicylic.acid.2.O.glucoside + salireposide + a.salicyloylsalicin + salicyloyl.coumaroyl.glucoside +
                    X2.5.dihydroxybenzoic.acid.methyl.ester.2.O.glucoside + HCH.deltoidin + tremulacin + HCH.salicortin + caffeoyltremuloidin +
                    salicyl.alcohol + salicylic.acid + salicortin, data = as.data.frame(metabolites), scale = T)
set.seed(01221990)
anova(x)




set.seed(01221990)
x <- vegan::dbrda(dist ~ ascorbic.acid.glucoside + malonic.acid + X14.70.284.269.glucoside + salicyl.salicylic.acid.2.O.glucoside + X14.85.168.glycoside + 
                      HCH.deltoidin + salicin + X1.2.4.benzenetriol + myo.inositol + catechol + 
                      salicyloyl.coumaroyl.glucoside + X2.5.dihydroxybenzoic.acid.methyl.ester.5.O.glucoside + gentisic.acid.2.O.glucoside + 
                      gentisic.acid.5.O.glucoside + X3.methoxysalicylic.acid.B.D.glucoside + vanillyl.alcohol.4.O.glucoside + salireposide + 
                      X16.25.297.583 + X1.3.dihydroxyacetone + fructose,
                  data = as.data.frame(metabolites_16S), scale = T)
set.seed(01221990)
anova(x)

set.seed(01221990)
x <- vegan::dbrda(dist ~ X14.70.284.269.glucoside + HCH.deltoidin + X17.16.363.273.257 + X9.82.269.284 + X17.06.259.glycoside + myo.inositol + X11.76.272.476.461 +
                    helicin + X14.70.405.315.268 + salicin + salicyl.alcohol + X19.19.451.395.587.602.512 + X14.85.168.glycoside + 
                    ascorbic.acid.glucoside + X1.2.dihydroxycyclohexanediol.glucoside + X16.15.297.361.209 + X1.2.4.benzenetriol + catechol + erythronic.acid +
                    pyrogallic.acid + X6.hydroxy.2.cyclohexenone.alcohol + malonic.acid + salicylic.acid + B.sitosterol + vanillic.acid + X2.5.dihydroxybenzoic.acid +
                    linoleic.acid + trans.p.coumaric.acid + a.linolenic.acid + X11.11.450.217.dehydrosugar + salicyloyl.coumaroyl.glucoside +
                    benzoic.acid + digalactosylglycerol + ferulic.acid + X15.96.297.lignan + purpurein + grandidentatin + X3.methoxysalicylic.acid.B.D.glucoside +
                    X16.05.297.lignan + X11.55.379.289 + salireposide + tremuloidin + monogalactosylglycerol + benzyl.alcohol + vanillyl.alcohol.4.O.glucoside +
                    quinic.acid + trichocarpinene + X2.5.dihydroxybenzoic.acid.methyl.ester.5.O.glucoside + X15.56.393.209.643.628.586.571.496.288 +
                    gentisic.acid.2.O.glucoside + caffeoyltremuloidin + trichocarpin + tremulacin,
                  data = as.data.frame(metabolites_16S), scale = T)
set.seed(01221990)
anova(x)

set.seed(01221990)
x <- vegan::dbrda(dist ~ salicin + salicyl.alcohol + salicylic.acid + salicortin, data = as.data.frame(metabolites_16S), scale = T)
set.seed(01221990)
anova(x)
set.seed(01221990)
anova(x, by = "mar") 

dist <- distance(data_16S_tss_RE, "bray")

set.seed(01221990)
x <- vegan::dbrda(dist ~ salicylaldehyde + trichocarpinene + helicin + iso.salicin + salirepin + gentisyl.alcohol.5.O.glucoside + salicin + gentisic.acid.2.O.glucoside +
                    gentisic.acid.5.O.glucoside + X2.5.dihydroxybenzoic.acid.methyl.ester.5.O.glucoside + X3.methoxysalicylic.acid.B.D.glucoside +
                    tremuloidin + trichocarpin + salicyl.salicylic.acid.2.O.glucoside + salireposide + a.salicyloylsalicin + salicyloyl.coumaroyl.glucoside +
                    X2.5.dihydroxybenzoic.acid.methyl.ester.2.O.glucoside + HCH.deltoidin + tremulacin + HCH.salicortin + caffeoyltremuloidin +
                    salicyl.alcohol + salicylic.acid + salicortin, data = as.data.frame(metabolites), scale = T)
set.seed(01221990)
anova(x)
set.seed(01221990)
anova(x, by = "mar") 

plot.data <- merge(summary(x)$sites, 
                   data.frame(sample_data(data_16S_tss_RE)), 
                   by = "row.names") 
ggplot() +
    geom_point(data = plot.data, aes(x = dbRDA1, y = dbRDA2, fill = resistance, shape = as.character(Infection_2020)), 
               size = 3) +
    scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604"), labels = c("deltoides", "trichocarpa - tolerant", "trichocarpa - susceptible")) +
    scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
    scale_x_continuous(name = paste0(paste0("dbRDA 1 (", round(100*as.data.frame(summary(x)$cont)[2,1], 1)), "%)")) +
    scale_y_continuous(name = paste0(paste0("dbRDA 2 (", round(100*as.data.frame(summary(x)$cont)[2,2], 1)), "%)")) +
    geom_segment(data = as.data.frame(summary(x)$biplot) %>% rownames_to_column() %>% filter(rowname == "salicortin" | rowname == "salicyl.alcohol"), 
                 aes(x = 0, y = 0, xend = dbRDA1*4, yend = dbRDA2*4),
                 size = 0.25, color = "black", arrow = arrow(length = unit(0.2, "cm"))) +
    geom_label_repel(data = as.data.frame(summary(x)$biplot) %>% 
                       rownames_to_column() %>% 
                       filter(rowname == "salicortin" | rowname == "salicyl.alcohol") %>%
                       mutate(rowname = dplyr::recode(rowname, salicyl.alcohol = "salicyl alcohol")),
                    aes(x = dbRDA1*4, y = dbRDA2*4, label = rowname), 
                    size = 4, alpha = 0.67) +
  theme(legend.position = "bottom", legend.title = element_blank(), legend.box = "vertical") +
  guides(fill = guide_legend(override.aes = list(pch = 21, size = 4)))-> p

ggsave("~/Desktop/BC_Septoria_Study/Figures/FigureS5.pdf", width = 6.5, height = 6.5, units = "in", dpi = 600)
p
dev.off()

dist <- distance(data_ITS_tss_RE, "bray")
set.seed(01221990)
x <- vegan::dbrda(dist ~ X19.19.451.395.587.602.512 + HCH.deltoidin + X17.06.259.glycoside + X17.16.363.273.257 + X14.70.284.269.glucoside + pyrogallic.acid + 
                    X9.82.269.284 + maleic.acid + myo.inositol + X14.85.168.glycoside + grandidentatin + salicortin + X14.00.292.375.204 + purpurein +
                    gentisic.acid.5.O.glucoside + salicyloyl.coumaroyl.glucoside + benzyl.glucoside + X16.32.344.glycoside + X3.methoxysalicylic.acid.B.D.glucoside +
                    raffinose,
                  data = as.data.frame(metabolites_ITS), scale = T)
set.seed(01221990)
anova(x)

set.seed(01221990)
x <- vegan::dbrda(dist ~ X14.70.284.269.glucoside + HCH.deltoidin + X17.16.363.273.257 + X9.82.269.284 + X17.06.259.glycoside + myo.inositol + X11.76.272.476.461 +
                    helicin + X14.70.405.315.268 + salicin + salicyl.alcohol + X19.19.451.395.587.602.512 + X14.85.168.glycoside + 
                    ascorbic.acid.glucoside + X1.2.dihydroxycyclohexanediol.glucoside + X16.15.297.361.209 + X1.2.4.benzenetriol + catechol + erythronic.acid +
                    pyrogallic.acid + X6.hydroxy.2.cyclohexenone.alcohol + malonic.acid + salicylic.acid + B.sitosterol + vanillic.acid + X2.5.dihydroxybenzoic.acid +
                    linoleic.acid + trans.p.coumaric.acid + a.linolenic.acid + X11.11.450.217.dehydrosugar + salicyloyl.coumaroyl.glucoside +
                    benzoic.acid + digalactosylglycerol + ferulic.acid + X15.96.297.lignan + purpurein + grandidentatin + X3.methoxysalicylic.acid.B.D.glucoside +
                    X16.05.297.lignan + X11.55.379.289 + salireposide + tremuloidin + monogalactosylglycerol + benzyl.alcohol + vanillyl.alcohol.4.O.glucoside +
                    quinic.acid + trichocarpinene + X2.5.dihydroxybenzoic.acid.methyl.ester.5.O.glucoside + X15.56.393.209.643.628.586.571.496.288 +
                    gentisic.acid.2.O.glucoside + caffeoyltremuloidin + trichocarpin + tremulacin,
                  data = as.data.frame(metabolites_ITS), scale = T)
set.seed(01221990)
anova(x)

set.seed(01221990)
x <- vegan::dbrda(dist ~ salicin + salicyl.alcohol + salicylic.acid + salicortin, data = as.data.frame(metabolites_ITS), scale = T)
set.seed(01221990)
anova(x)

```


# Microbiome 3 (Differential Abundance)
```{r} 
Phylum_16S <- data_16S_RE_counts %>% tax_glom("Phylum") 
out_16S_phylum <- ancombc(phyloseq = Phylum_16S, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
res <- out_16S_phylum$res 

dif.abun.only <- res$diff_abn %>% filter(Copy_number_per_ul == TRUE)

da <- merge(dif.abun.only, res$beta, by = 0, all.x = T)
da <- merge(da, res$se, by.x = "Row.names", by.y = 0, all.x = T)
da <- merge(da, Phylum_16S@tax_table, by.x = "Row.names", by.y = 0, all.x = T)

da$Phylum <- sapply(strsplit(as.character(da$Phylum), "__"), `[`, 2)

da_16S <- da %>% 
  mutate(Phylum = fct_reorder(Phylum, as.numeric(Copy_number_per_ul.y))) %>%
  ggplot(aes(x = Phylum, y = Copy_number_per_ul.y, fill = Phylum)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Copy_number_per_ul.y - Copy_number_per_ul, ymax = Copy_number_per_ul.y + Copy_number_per_ul), width = 0.2) +
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log Fold Change Copy #"^-1)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1))

Class_ITS <- data_ITS_RE_counts %>% tax_glom("Class") 
out_ITS_class <- ancombc(phyloseq = Class_ITS, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
res <- out_ITS_class$res 

dif.abun.only <- res$diff_abn %>% filter(Copy_number_per_ul == TRUE)

da <- merge(dif.abun.only, res$beta, by = 0, all.x = T)
da <- merge(da, res$se, by.x = "Row.names", by.y = 0, all.x = T)
da <- merge(da, Class_ITS@tax_table, by.x = "Row.names", by.y = 0, all.x = T)

da$Class <- sapply(strsplit(as.character(da$Class), "__"), `[`, 2)

da_ITS <- da %>% 
  mutate(Class = fct_reorder(Class, as.numeric(Copy_number_per_ul.y))) %>%
  ggplot(aes(x = Class, y = Copy_number_per_ul.y, fill = Class)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Copy_number_per_ul.y - Copy_number_per_ul, ymax = Copy_number_per_ul.y + Copy_number_per_ul), width = 0.2) +
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log Fold Change Copy #"^-1)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1))

prow <- plot_grid(da_16S, da_ITS, ncol = 2, align = "vh", labels = "AUTO")

ggsave("~/Desktop/BC_Septoria_Study/Figures/Figure5.pdf", width = 7.5, height = 5, units = "in", dpi = 600)
prow
dev.off()

# without septoria

Class_ITS <- data_ITS_RE_counts %>% subset_taxa(Genus != "g__Septoria") %>% tax_glom("Class") 
out_ITS_class <- ancombc(phyloseq = Class_ITS, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
res <- out_ITS_class$res 

dif.abun.only <- res$diff_abn %>% filter(Copy_number_per_ul == TRUE)

da <- merge(dif.abun.only, res$beta, by = 0, all.x = T)
da <- merge(da, res$se, by.x = "Row.names", by.y = 0, all.x = T)
da <- merge(da, Class_ITS@tax_table, by.x = "Row.names", by.y = 0, all.x = T)

da$Class <- sapply(strsplit(as.character(da$Class), "__"), `[`, 2)

da_ITS <- da %>% 
  mutate(Class = fct_reorder(Class, as.numeric(Copy_number_per_ul.y))) %>%
  ggplot(aes(x = Class, y = Copy_number_per_ul.y, fill = Class)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Copy_number_per_ul.y - Copy_number_per_ul, ymax = Copy_number_per_ul.y + Copy_number_per_ul), width = 0.2) +
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log Fold Change Copy #"^-1)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1))




Phylum_16S <- data_16S_RH_counts %>% tax_glom("Phylum") 
out_16S_phylum <- ancombc(phyloseq = Phylum_16S, formula = "Copy_number_per_ul", 
                          p_adj_method = "holm", zero_cut = 0.90, lib_cut = 1000, tol = 1e-5, 
                          max_iter = 100, alpha = 0.05)
res <- out_16S_phylum$res 

dif.abun.only <- res$diff_abn %>% filter(Copy_number_per_ul == TRUE)

da <- merge(dif.abun.only, res$beta, by = 0, all.x = T)
da <- merge(da, res$se, by.x = "Row.names", by.y = 0, all.x = T)
da <- merge(da, Phylum_16S@tax_table, by.x = "Row.names", by.y = 0, all.x = T)

da$Phylum <- sapply(strsplit(as.character(da$Phylum), "__"), `[`, 2)

da_16S <- da %>% 
  mutate(Phylum = fct_reorder(Phylum, as.numeric(Copy_number_per_ul.y))) %>%
  ggplot(aes(x = Phylum, y = Copy_number_per_ul.y, fill = Phylum)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Copy_number_per_ul.y - Copy_number_per_ul, ymax = Copy_number_per_ul.y + Copy_number_per_ul), width = 0.2) +
  scale_x_discrete(name = NULL) +
  scale_y_continuous(name = bquote("Log Fold Change Copy #"^-1)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1))


ggsave("~/Desktop/BC_Septoria_Study/MAC_presentation/root_ancom.pdf", width = 11, height = 6, units = "in", dpi = 600)
prow
dev.off()
```





# Funguild
```{r}
#fg <- parse_funguild() #Accessed 2/25/21
#data_ITS_tax <- as.data.frame(as(tax_table(data_ITS_combined), "matrix"))
#data_ITS_tax$Genus <- sapply(strsplit(as.character(data_ITS_tax$Genus), "__"), `[`, 2)
#data_ITS_tax$OTU <- rownames(data_ITS_tax)
#data_ITS_tax_fg <- merge(data_ITS_tax, 
#                         fg %>% filter(taxonomicLevel == "Genus"), 
#                         by.x = "Genus", by.y = "taxon", all.x = T)
#data_ITS_tax_fg$guild[str_detect(data_ITS_tax_fg$guild, "Ectomycorrhizal") == TRUE] <- "Ectomycorrhizal"
#data_ITS_tax_fg$guild[data_ITS_tax_fg$Phylum == "p__Glomeromycota"] <- "Arbuscular Mycorrhizal"
#data_ITS_tax_fg$guild[str_detect(data_ITS_tax_fg$guild, "athogen") == TRUE] <- "Pathogen"
#data_ITS_tax_fg$guild[str_detect(data_ITS_tax_fg$guild, "Saprotroph") == TRUE] <- "Saprotroph"
#
#write.csv(data_ITS_tax_fg, "~/Desktop/BC_Septoria_Study/Data/ITS/FUNGuild_tax_table.csv")
data_ITS_tax_fg <- read.csv("~/Desktop/BC_Septoria_Study/Data/ITS/FUNGuild_tax_table.csv")
TAX <- tax_table(data_ITS_tax_fg %>% dplyr::select("guild", "Phylum", "Class", "Order", "Family", "Genus", "Species", "OTU", "confidenceRanking"))
taxa_names(TAX) <- data_ITS_tax_fg$OTU
data_ITS_fg <- merge_phyloseq(otu_table(data_ITS_combined), TAX, data_ITS_combined@sam_data)

data_ITS_fg_glom <- tax_glom(data_ITS_fg, "ta1")

data_ITS_fg_tss <- transform_sample_counts(data_ITS_fg_glom, function(x) x/sum(x))
myco <- data_ITS_fg_tss %>% subset_taxa(ta1 == "Ectomycorrhizal" | ta1 == "Arbuscular Mycorrhizal")
mycomelt <- psmelt(myco)

em <- filter(mycomelt, ta1 == "Ectomycorrhizal")
am <- filter(mycomelt, ta1 == "Arbuscular Mycorrhizal")

mycomelt %>% filter(sample_type == "RH") %>%
  ggplot(aes(x = species, y = Abundance *100, fill = species)) +
  geom_boxplot() +
  scale_y_continuous(name = "Relative Abundnace (%)") +
  scale_x_discrete(name = NULL) +
  facet_wrap(~ta1, scales = "free") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
  theme(axis.text.x = element_text(face = "italic")) -> p

ggsave("~/Desktop/BC_Septoria_Study/MAC_presentation/mycorrhizal.pdf", width = 7, height = 6, units = "in", dpi = 600)
p
dev.off()


y.transf.betareg <- function(y){
  n.obs <- sum(!is.na(y))
  (y * (n.obs - 1) + 0.5) / n.obs
}
x <- betareg(y.transf.betareg(Abundance) ~ species, am %>% filter(sample_type == "RH"))
summary(x)
joint_tests(x) 

x <- betareg(y.transf.betareg(Abundance) ~ species*sample_type, am)
summary(x)
joint_tests(x) 


x <- betareg(y.transf.betareg(Abundance) ~ Infection_2020*sample_type, am)
summary(x)
joint_tests(x) 



```

```{r}
# rare taxa ---------------------------------------

data_16S_tss <- microbiome::transform(data_16S_combined, transform = "compositional")
data_ITS_tss <- microbiome::transform(data_ITS_combined, transform = "compositional")




RE_16S <- data_16S_tss %>% 
  subset_samples(sample_type == "RE") %>%
  tax_glom("Genus")

rare_RE_16S <- data_16S_tss %>% 
  subset_samples(sample_type == "RE") %>%
  tax_glom("Genus") %>%
  filter_taxa(function(x) mean(x) < 0.01, TRUE) %>%
  microbiome::transform(transform = "compositional")

proteos <- subset_taxa(RE_16S, Class == "D_2__Alphaproteobacteria" |
                               Class == "D_2__Betaproteobacteria" |
                               Class == "D_2__Deltaproteobacteria" |
                               Class == "D_2__Gammaproteobacteria")
colnames(tax_table(proteos))[1] <- "group"
tax_table(proteos)[,"group"] <- tax_table(proteos)[,"Class"]
proteos_glom <- tax_glom(proteos, taxrank = "group")
proteos_melt <- psmelt(proteos_glom)

#Do same for Archaea

archaea <- subset_taxa(RE_16S, Kingdom == "D_0__Archaea")
colnames(tax_table(archaea))[1] <- "group"
archaea_glom <- tax_glom(archaea, taxrank = "group")
archaea_melt <- psmelt(archaea_glom)

#Get the rest of the abundant phyla, assign grouping by phylum, glom, and melt
abundant_non_proteos <- subset_taxa(RE_16S, Phylum == "D_1__Verrucomicrobia" |
                                            Phylum == "D_1__TM7"|
                                            Phylum == "D_1__Planctomycetes"|
                                            Phylum == "D_1__Firmicutes" |
                                            Phylum == "D_1__Gemmatimonadetes" |
                                            Phylum == "D_1__Chloroflexi" |
                                            Phylum == "D_1__Bacteroidetes" |
                                            Phylum == "D_1__Acidobacteria" |
                                            Phylum == "D_1__Actinobacteria")
colnames(tax_table(abundant_non_proteos))[1] <- "group"
tax_table(abundant_non_proteos)[,"group"] <- tax_table(abundant_non_proteos)[,"Phylum"]
abundant_non_proteos_glom <- tax_glom(abundant_non_proteos, taxrank = "group")
abundant_non_proteos_melt <- psmelt(abundant_non_proteos_glom)

#concatenate the datasets
data_16S_grouping <- rbind(proteos_melt, abundant_non_proteos_melt, archaea_melt)

#summarise averages of time x niche x group combos
sum <- data_16S_grouping %>% 
  group_by(species, group) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$group <- sapply(strsplit(as.character(sum$group), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(group = "Other")

#concatenate the datasets
sum = rbind(sum, rare)

#order groups
sum$group <- ordered(sum$group, c("Archaea", "Alphaproteobacteria", "Betaproteobacteria",
                                  "Deltaproteobacteria", "Gammaproteobacteria", "Acidobacteria",
                                  "Actinobacteria", "Bacteroidetes", "Chloroflexi", "Firmicutes",
                                  "Gemmatimonadetes", "Planctomycetes", "TM7", "Verrucomicrobia",
                                  "Other"))

rel_16S <- ggplot(sum, aes(x = species, y = means, fill = group)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values = c("#ff7f0e", "#0e334d","#1f77b4", "#5ca7db", "#abd2ed", "yellow",
                                 "#2ca02c", "#d62728", "#9467bd", "#8c564b", "#e377c2", 
                                 "gray", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "All") 





proteos <- subset_taxa(rare_RE_16S, Class == "D_2__Alphaproteobacteria" |
                               Class == "D_2__Betaproteobacteria" |
                               Class == "D_2__Deltaproteobacteria" |
                               Class == "D_2__Gammaproteobacteria")
colnames(tax_table(proteos))[1] <- "group"
tax_table(proteos)[,"group"] <- tax_table(proteos)[,"Class"]
proteos_glom <- tax_glom(proteos, taxrank = "group")
proteos_melt <- psmelt(proteos_glom)

#Do same for Archaea

archaea <- subset_taxa(rare_RE_16S, Kingdom == "D_0__Archaea")
colnames(tax_table(archaea))[1] <- "group"
archaea_glom <- tax_glom(archaea, taxrank = "group")
archaea_melt <- psmelt(archaea_glom)

#Get the rest of the abundant phyla, assign grouping by phylum, glom, and melt
abundant_non_proteos <- subset_taxa(rare_RE_16S, Phylum == "D_1__Verrucomicrobia" |
                                            Phylum == "D_1__TM7"|
                                            Phylum == "D_1__Planctomycetes"|
                                            Phylum == "D_1__Firmicutes" |
                                            Phylum == "D_1__Gemmatimonadetes" |
                                            Phylum == "D_1__Chloroflexi" |
                                            Phylum == "D_1__Bacteroidetes" |
                                            Phylum == "D_1__Acidobacteria" |
                                            Phylum == "D_1__Actinobacteria")
colnames(tax_table(abundant_non_proteos))[1] <- "group"
tax_table(abundant_non_proteos)[,"group"] <- tax_table(abundant_non_proteos)[,"Phylum"]
abundant_non_proteos_glom <- tax_glom(abundant_non_proteos, taxrank = "group")
abundant_non_proteos_melt <- psmelt(abundant_non_proteos_glom)

#concatenate the datasets
data_16S_grouping <- rbind(proteos_melt, abundant_non_proteos_melt, archaea_melt)

#summarise averages of time x niche x group combos
sum <- data_16S_grouping %>% 
  group_by(species, group) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$group <- sapply(strsplit(as.character(sum$group), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(group = "Other")

#concatenate the datasets
sum = rbind(sum, rare)

#order groups
sum$group <- ordered(sum$group, c("Archaea", "Alphaproteobacteria", "Betaproteobacteria",
                                  "Deltaproteobacteria", "Gammaproteobacteria", "Acidobacteria",
                                  "Actinobacteria", "Bacteroidetes", "Chloroflexi", "Firmicutes",
                                  "Gemmatimonadetes", "Planctomycetes", "TM7", "Verrucomicrobia",
                                  "Other"))

rare_rel_16S <- ggplot(sum, aes(x = species, y = means, fill = group)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values = c("#ff7f0e", "#0e334d","#1f77b4", "#5ca7db", "#abd2ed", "yellow",
                                 "#2ca02c", "#d62728", "#9467bd", "#8c564b", "#e377c2", 
                                 "grey", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "Rare")





RH_16S <- data_16S_tss %>% 
  subset_samples(sample_type == "RH") %>%
  tax_glom("Genus")

rare_RH_16S <- data_16S_tss %>% 
  subset_samples(sample_type == "RH") %>%
  tax_glom("Genus") %>%
  filter_taxa(function(x) mean(x) < 0.01, TRUE) %>%
  microbiome::transform(transform = "compositional")

proteos <- subset_taxa(RH_16S, Class == "D_2__Alphaproteobacteria" |
                               Class == "D_2__Betaproteobacteria" |
                               Class == "D_2__Deltaproteobacteria" |
                               Class == "D_2__Gammaproteobacteria")
colnames(tax_table(proteos))[1] <- "group"
tax_table(proteos)[,"group"] <- tax_table(proteos)[,"Class"]
proteos_glom <- tax_glom(proteos, taxrank = "group")
proteos_melt <- psmelt(proteos_glom)

#Do same for Archaea

archaea <- subset_taxa(RH_16S, Kingdom == "D_0__Archaea")
colnames(tax_table(archaea))[1] <- "group"
archaea_glom <- tax_glom(archaea, taxrank = "group")
archaea_melt <- psmelt(archaea_glom)

#Get the rest of the abundant phyla, assign grouping by phylum, glom, and melt
abundant_non_proteos <- subset_taxa(RH_16S, Phylum == "D_1__Verrucomicrobia" |
                                            Phylum == "D_1__TM7"|
                                            Phylum == "D_1__Planctomycetes"|
                                            Phylum == "D_1__Firmicutes" |
                                            Phylum == "D_1__Gemmatimonadetes" |
                                            Phylum == "D_1__Chloroflexi" |
                                            Phylum == "D_1__Bacteroidetes" |
                                            Phylum == "D_1__Acidobacteria" |
                                            Phylum == "D_1__Actinobacteria")
colnames(tax_table(abundant_non_proteos))[1] <- "group"
tax_table(abundant_non_proteos)[,"group"] <- tax_table(abundant_non_proteos)[,"Phylum"]
abundant_non_proteos_glom <- tax_glom(abundant_non_proteos, taxrank = "group")
abundant_non_proteos_melt <- psmelt(abundant_non_proteos_glom)

#concatenate the datasets
data_16S_grouping <- rbind(proteos_melt, abundant_non_proteos_melt, archaea_melt)

#summarise averages of time x niche x group combos
sum <- data_16S_grouping %>% 
  group_by(species, group) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$group <- sapply(strsplit(as.character(sum$group), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(group = "Other")

#concatenate the datasets
sum = rbind(sum, rare)

#order groups
sum$group <- ordered(sum$group, c("Archaea", "Alphaproteobacteria", "Betaproteobacteria",
                                  "Deltaproteobacteria", "Gammaproteobacteria", "Acidobacteria",
                                  "Actinobacteria", "Bacteroidetes", "Chloroflexi", "Firmicutes",
                                  "Gemmatimonadetes", "Planctomycetes", "TM7", "Verrucomicrobia",
                                  "Other"))

rel_16S_RH <- ggplot(sum, aes(x = species, y = means, fill = group)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values = c("#ff7f0e", "#0e334d","#1f77b4", "#5ca7db", "#abd2ed", "yellow",
                                 "#2ca02c", "#d62728", "#9467bd", "#8c564b", "#e377c2", 
                                 "gray", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "All") 





proteos <- subset_taxa(rare_RH_16S, Class == "D_2__Alphaproteobacteria" |
                               Class == "D_2__Betaproteobacteria" |
                               Class == "D_2__Deltaproteobacteria" |
                               Class == "D_2__Gammaproteobacteria")
colnames(tax_table(proteos))[1] <- "group"
tax_table(proteos)[,"group"] <- tax_table(proteos)[,"Class"]
proteos_glom <- tax_glom(proteos, taxrank = "group")
proteos_melt <- psmelt(proteos_glom)

#Do same for Archaea

archaea <- subset_taxa(rare_RH_16S, Kingdom == "D_0__Archaea")
colnames(tax_table(archaea))[1] <- "group"
archaea_glom <- tax_glom(archaea, taxrank = "group")
archaea_melt <- psmelt(archaea_glom)

#Get the rest of the abundant phyla, assign grouping by phylum, glom, and melt
abundant_non_proteos <- subset_taxa(rare_RH_16S, Phylum == "D_1__Verrucomicrobia" |
                                            Phylum == "D_1__TM7"|
                                            Phylum == "D_1__Planctomycetes"|
                                            Phylum == "D_1__Firmicutes" |
                                            Phylum == "D_1__Gemmatimonadetes" |
                                            Phylum == "D_1__Chloroflexi" |
                                            Phylum == "D_1__Bacteroidetes" |
                                            Phylum == "D_1__Acidobacteria" |
                                            Phylum == "D_1__Actinobacteria")
colnames(tax_table(abundant_non_proteos))[1] <- "group"
tax_table(abundant_non_proteos)[,"group"] <- tax_table(abundant_non_proteos)[,"Phylum"]
abundant_non_proteos_glom <- tax_glom(abundant_non_proteos, taxrank = "group")
abundant_non_proteos_melt <- psmelt(abundant_non_proteos_glom)

#concatenate the datasets
data_16S_grouping <- rbind(proteos_melt, abundant_non_proteos_melt, archaea_melt)

#summarise averages of time x niche x group combos
sum <- data_16S_grouping %>% 
  group_by(species, group) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$group <- sapply(strsplit(as.character(sum$group), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(group = "Other")

#concatenate the datasets
sum = rbind(sum, rare)

#order groups
sum$group <- ordered(sum$group, c("Archaea", "Alphaproteobacteria", "Betaproteobacteria",
                                  "Deltaproteobacteria", "Gammaproteobacteria", "Acidobacteria",
                                  "Actinobacteria", "Bacteroidetes", "Chloroflexi", "Firmicutes",
                                  "Gemmatimonadetes", "Planctomycetes", "TM7", "Verrucomicrobia",
                                  "Other"))

rare_rel_16S_RH <- ggplot(sum, aes(x = species, y = means, fill = group)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values = c("#ff7f0e", "#0e334d","#1f77b4", "#5ca7db", "#abd2ed", "yellow",
                                 "#2ca02c", "#d62728", "#9467bd", "#8c564b", "#e377c2", 
                                 "grey", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "Rare")

x <- plot_grid(rel_16S + theme(legend.position = "none"), rare_rel_16S + theme(legend.position = "none"),
               rel_16S_RH + theme(legend.position = "none"), rare_rel_16S_RH + theme(legend.position = "none"), ncol = 4)

leg <- get_legend(rare_rel_16S_RH)

x2 <- plot_grid(x, leg, rel_widths = c(1, 0.25), ncol = 2)

rare_RE_16S <- data_16S_tss %>% 
  subset_samples(sample_type == "RE") %>%
  tax_glom("Genus") %>%
  filter_taxa(function(x) mean(x) < 0.01, TRUE) %>%
  otu_table %>%
  sum

data_ITS_tss %>% 
  subset_samples(sample_type == "RE") %>%
  tax_glom("Genus") %>%
  filter_taxa(function(x) mean(x) < 0.01, TRUE) %>%
  otu_table %>%
  sum

RE_ITS <- data_ITS_tss %>% 
  subset_samples(sample_type == "RE") %>%
  tax_glom("Genus") %>%
  microbiome::transform(transform = "compositional")

rare_RE_ITS <- data_ITS_tss %>% 
  subset_samples(sample_type == "RE") %>%
  tax_glom("Genus") %>%
  filter_taxa(function(x) mean(x) < 0.01, TRUE) %>%
  microbiome::transform(transform = "compositional")


order_sum_ITS <- tapply(taxa_sums(data_ITS_tss), 
                         tax_table(data_ITS_tss)[, "Phylum"], sum, na.rm=TRUE)
top10order_ITS <- names(sort(order_sum_ITS, TRUE))[1:10]

abundant_ITS_order <- subset_taxa(RE_ITS, Phylum %in% top10order_ITS)
abundant_ITS_order_glom <- tax_glom(abundant_ITS_order, taxrank = "Phylum")
abundant_ITS_order_melt <- psmelt(abundant_ITS_order_glom)

#summarise averages of time x niche x group combos
sum <- abundant_ITS_order_melt %>% 
  group_by(species, Phylum) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$Phylum <- sapply(strsplit(as.character(sum$Phylum), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Other")

#concatenate the datasets
sum <- rbind(sum, rare)

sum$Phylum <- forcats::fct_relevel(sum$Phylum, "Other", after = Inf)

rel_ITS_RE <- ggplot(sum, aes(x = species, y = means, fill = Phylum)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values =c("#ff7f0e", "#0e334d","#1f77b4", "#abd2ed", "yellow",
                                 "#8c564b", "#d62728", "#9467bd", "#2ca02c", "#e377c2", 
                                 "#7f7f7f", "#17becf", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "All") 


abundant_ITS_order <- subset_taxa(rare_RE_ITS, Phylum %in% top10order_ITS)
abundant_ITS_order_glom <- tax_glom(abundant_ITS_order, taxrank = "Phylum")
abundant_ITS_order_melt <- psmelt(abundant_ITS_order_glom)

#summarise averages of time x niche x group combos
sum <- abundant_ITS_order_melt %>% 
  group_by(species, Phylum) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$Phylum <- sapply(strsplit(as.character(sum$Phylum), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Other")

#concatenate the datasets
sum <- rbind(sum, rare)

sum$Phylum <- forcats::fct_relevel(sum$Phylum, "Other", after = Inf)

rel_ITS_RE_rare <- ggplot(sum, aes(x = species, y = means, fill = Phylum)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values =c("#ff7f0e", "#0e334d","#1f77b4", "#abd2ed", "yellow",
                                 "#8c564b", "#d62728", "#9467bd", "#2ca02c", "#e377c2", 
                                 "#7f7f7f", "#17becf", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "All") 

RH_ITS <- data_ITS_tss %>% 
  subset_samples(sample_type == "RH") %>%
  tax_glom("Genus") %>%
  microbiome::transform(transform = "compositional")

rare_RH_ITS <- data_ITS_tss %>% 
  subset_samples(sample_type == "RH") %>%
  tax_glom("Genus") %>%
  filter_taxa(function(x) mean(x) < 0.01, TRUE) %>%
  microbiome::transform(transform = "compositional")


abundant_ITS_order <- subset_taxa(RH_ITS, Phylum %in% top10order_ITS)
abundant_ITS_order_glom <- tax_glom(abundant_ITS_order, taxrank = "Phylum")
abundant_ITS_order_melt <- psmelt(abundant_ITS_order_glom)

#summarise averages of time x niche x group combos
sum <- abundant_ITS_order_melt %>% 
  group_by(species, Phylum) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$Phylum <- sapply(strsplit(as.character(sum$Phylum), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Other")

#concatenate the datasets
sum <- rbind(sum, rare)

sum$Phylum <- forcats::fct_relevel(sum$Phylum, "Other", after = Inf)

rel_ITS_RH <- ggplot(sum, aes(x = species, y = means, fill = Phylum)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values =c("#ff7f0e", "#0e334d","#1f77b4", "#abd2ed", "yellow",
                                 "#8c564b", "#d62728", "#9467bd", "#2ca02c", "#e377c2", 
                                 "#7f7f7f", "#17becf", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "All") 


abundant_ITS_order <- subset_taxa(rare_RH_ITS, Phylum %in% top10order_ITS)
abundant_ITS_order_glom <- tax_glom(abundant_ITS_order, taxrank = "Phylum")
abundant_ITS_order_melt <- psmelt(abundant_ITS_order_glom)

#summarise averages of time x niche x group combos
sum <- abundant_ITS_order_melt %>% 
  group_by(species, Phylum) %>%
  dplyr::summarise(means = mean(Abundance))

#Fix names
sum$Phylum <- sapply(strsplit(as.character(sum$Phylum), "__"), `[`, 2)

#Everything that's left is considered rare  
rare <- sum %>%
  group_by(species) %>%
  dplyr::summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Other")

#concatenate the datasets
sum <- rbind(sum, rare)

sum$Phylum <- forcats::fct_relevel(sum$Phylum, "Other", after = Inf)

rel_ITS_RH_rare <- ggplot(sum, aes(x = species, y = means, fill = Phylum)) +
  geom_bar(position = "stack", stat = "identity", col = "black") +
  theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
  scale_fill_manual(values =c("#ff7f0e", "#0e334d","#1f77b4", "#abd2ed", "yellow",
                                 "#8c564b", "#d62728", "#9467bd", "#2ca02c", "#e377c2", 
                                 "#7f7f7f", "#17becf", "white")) +
  scale_y_continuous(name = NULL, breaks = NULL) +
  labs(x = NULL, title = "All") 

x <- plot_grid(rel_ITS_RE + theme(legend.position = "none"), rel_ITS_RE_rare + theme(legend.position = "none"),
               rel_ITS_RH + theme(legend.position = "none"), rel_ITS_RH_rare + theme(legend.position = "none"), ncol = 4)

leg <- get_legend(rel_ITS_RH)

x2 <- plot_grid(x, leg, rel_widths = c(1, 0.25), ncol = 2)






y <- data_16S_tss %>% subset_samples(sample_type == "RH")
y@otu_table[y@otu_table > 0] <- 1
mean(colSums(y@otu_table))

y <- data_ITS_tss %>% subset_samples(sample_type == "RH")
y@otu_table[y@otu_table > 0] <- 1
mean(colSums(y@otu_table))


y <- data_16S_tss %>% subset_samples(sample_type == "RH") %>% tax_glom("Genus") 
y@otu_table[y@otu_table > 0] <- 1
mean(colSums(y@otu_table))

y <- data_ITS_tss %>% subset_samples(sample_type == "RH") %>% tax_glom("Genus")
y@otu_table[y@otu_table > 0] <- 1
mean(colSums(y@otu_table))



rare_RH_16S <- data_16S_tss %>% 
  subset_samples(sample_type == "RH") %>%
  filter_taxa(function(x) mean(x) < 0.0001, TRUE)


rare_RE_ITS <- data_ITS_tss %>% 
  subset_samples(sample_type == "RE") %>%
  filter_taxa(function(x) mean(x) < 0.0001, TRUE)

rare_RH_ITS <- data_ITS_tss %>% 
  subset_samples(sample_type == "RH") %>%
  filter_taxa(function(x) mean(x) < 0.0001, TRUE)


rare_RE_16S@otu_table[rare_RE_16S@otu_table > 0] <- 1
rich_16S_RE <- as.data.frame(colSums(rare_RE_16S@otu_table))
rich_16S_RE <- merge(rich_16S_RE, rare_RE_16S@sam_data, by = 0)

rare_RH_16S@otu_table[rare_RH_16S@otu_table > 0] <- 1
rich_16S_RH <- as.data.frame(colSums(rare_RH_16S@otu_table))
rich_16S_RH <- merge(rich_16S_RH, rare_RH_16S@sam_data, by = 0)

rare_RE_ITS@otu_table[rare_RE_ITS@otu_table > 0] <- 1
rich_ITS_RE <- as.data.frame(colSums(rare_RE_ITS@otu_table))
rich_ITS_RE <- merge(rich_ITS_RE, rare_RE_ITS@sam_data, by = 0)

rare_RH_ITS@otu_table[rare_RH_ITS@otu_table > 0] <- 1
rich_ITS_RH <- as.data.frame(colSums(rare_RH_ITS@otu_table))
rich_ITS_RH <- merge(rich_ITS_RH, rare_RH_ITS@sam_data, by = 0)


plot_data <- rbind(rich_16S_RE %>% mutate(richness = `colSums(rare_RE_16S@otu_table)`, amplicon = "16S") %>%
             dplyr::select(richness, genotype, resistance, Infection_2020, sample_type, amplicon), 
           rich_16S_RH %>% mutate(richness = `colSums(rare_RH_16S@otu_table)`, amplicon = "16S") %>%
             dplyr::select(richness, genotype, resistance, Infection_2020, sample_type, amplicon),
           rich_ITS_RE %>% mutate(richness = `colSums(rare_RE_ITS@otu_table)`, amplicon = "ITS") %>%
             dplyr::select(richness, genotype, resistance, Infection_2020, sample_type, amplicon), 
           rich_ITS_RH %>% mutate(richness = `colSums(rare_RH_ITS@otu_table)`, amplicon = "ITS") %>%
             dplyr::select(richness, genotype, resistance, Infection_2020, sample_type, amplicon))

plot_data %>%
  dplyr::mutate(genotype = ordered(genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(genotype, resistance, sample_type, amplicon) %>%
  dplyr::summarise(mean = mean(richness), se = sd(richness)/sqrt(n())) %>%
  ggplot(aes(x = genotype, y = mean, fill = resistance)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#0D5F8A", "#B6D084", "#2F7604")) +
  geom_jitter(data = plot_data, aes(x = genotype, y = richness, fill = resistance, shape = as.factor(as.character(Infection_2020))), inherit.aes = F, alpha = 0.5) +
  scale_shape_manual(values = c(21, 24), labels = c("noninfected", "infected")) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  facet_grid(amplicon~sample_type, scales = "free", 
             labeller = labeller(sample_type = c("RE" = "Root Endosphere", "RH" = "Rhizosphere"),
                                 amplicon = c("16S" = "Archea & Bacteria", "ITS" = "Fungi"))) +
  scale_y_continuous(name = "Richness") +
  panel_border() +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust = 1), legend.title = element_blank(), axis.text.y = element_text(size = 18),
        legend.position = "right", legend.text = element_text(size = 20), axis.title.x = element_blank(), axis.title.y = element_text(size = 20),
        strip.text.y = element_text(size = 18), strip.text.x = element_text(size = 18)) -> p

ggsave("~/Desktop/BC_Septoria_Study/MAC_presentation/rare_richness.pdf", width = 11, height = 6, units = "in", dpi = 600)
p
dev.off() 


plot_data %>%
  dplyr::mutate(genotype = ordered(genotype, 
                                   c("D411", "D176", "GW9823", "BESC835", "BESC261", "BESC22", "BESC801", "BESC71", "BESC103", "BESC148"))) %>%
  group_by(genotype, resistance, sample_type, amplicon, Infection_2020) %>%
  dplyr::summarise(mean = mean(richness), se = sd(richness)/sqrt(n()), n = n()) -> x

write.csv(x, "~/Desktop/BC_Septoria_Study/MAC_presentation/rare_rich_means.csv")

write.csv(plot_data, "~/Desktop/BC_Septoria_Study/MAC_presentation/rare_rich_full.csv")

```



